<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />



<meta name="date" content="2024-12-11" />

<title>SPEAQeasy bootcamp</title>

<script src="site_libs/header-attrs-2.6/header-attrs.js"></script>
<script src="site_libs/jquery-3.5.1/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/htmlwidgets-1.5.3/htmlwidgets.js"></script>
<script src="site_libs/plotly-binding-4.9.3/plotly.js"></script>
<script src="site_libs/typedarray-0.1/typedarray.min.js"></script>
<link href="site_libs/crosstalk-1.1.1/css/crosstalk.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.1.1/js/crosstalk.min.js"></script>
<link href="site_libs/plotly-htmlwidgets-css-1.57.1/plotly-htmlwidgets.css" rel="stylesheet" />
<script src="site_libs/plotly-main-1.57.1/plotly-latest.min.js"></script>
<link href="site_libs/selectize-0.12.0/selectize.bootstrap3.css" rel="stylesheet" />
<script src="site_libs/selectize-0.12.0/selectize.min.js"></script>
<link href="site_libs/colourpicker-1.1/colourpicker.min.css" rel="stylesheet" />
<script src="site_libs/colourpicker-1.1/colourpicker.min.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">

</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


<link rel="stylesheet" href="/__w/_temp/Library/BiocStyle/resources/html/bioconductor.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 828px;
  margin-left: auto;
  margin-right: auto;
}

img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {

}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 246px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



<script>
function toggle_visibility(id1) {
  var e = document.getElementById(id1);
  e.style.display = ((e.style.display!="none") ? "none" : "block");
}
</script>

</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">SPEAQeasy example</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="prepare_data.html">Download Example Data</a>
</li>
<li>
  <a href="run_speaqeasy.html">Run SPEAQeasy</a>
</li>
<li>
  <a href="swap_speaqeasy.html">Resolve Sample Identities</a>
</li>
<li>
  <a href="de_analysis_speaqeasy.html">Differential Expression</a>
</li>
<li>
  <a href="bootcamp_intro">Bootcamp</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/LieberInstitute/SPEAQeasy-example">
    <span class="fas fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">SPEAQeasy bootcamp</h1>
<p class="author-name">Leonardo Collado-Torres<span class="affil-mark">1,2*</span></p>
<p class="author-affiliation"><span class="affil-mark">1</span>Lieber Institute for Brain Development, Johns Hopkins Medical Campus<br><span class="affil-mark">2</span>Center for Computational Biology, Johns Hopkins University</p>
<p class="author-email"><span class="affil-mark">*</span><a href="mailto:lcolladotor@gmail.com">lcolladotor@gmail.com</a></p>
<h4 class="date">11 December 2024</h4>

</div>


<p>This document is part of the <em>R/Bioconductor-powered Team Data Science</em> <a href="https://lcolladotor.github.io/bioc_team_ds/rbioconductor-data-science-bootcamps.html#libd-bootcamps">LIBD bootcamps</a>.</p>
<p>You can download the <a href="https://github.com/LieberInstitute/SPEAQeasy-example/blob/master/bootcamp_intro.Rmd"><strong>latest version of this Rmd document here</strong></a>. <label for="sidenote-1" class="sidenote-number">1</label><label for="sidenote-1" class="margin-toggle" onclick="toggle_visibility('sidenote-1')">1</label><span class="sidenote" id="sidenote-1"><span class="sidenote-number">1</span> This was useful particularly during October 5-7 while I was updating this document frequently.</span></p>
<div id="setup" class="section level1" number="1">
<h1><span class="header-section-number">1</span> Setup</h1>
<p>Please install the latest:</p>
<ul>
<li>R version from <a href="https://cran.r-project.org/">CRAN</a></li>
<li><a href="https://rstudio.com/products/rstudio/download/">RStudio Destkop</a> <label for="sidenote-2" class="sidenote-number">2</label><label for="sidenote-2" class="margin-toggle" onclick="toggle_visibility('sidenote-2')">2</label><span class="sidenote" id="sidenote-2"><span class="sidenote-number">2</span> You might want to try out <a href="https://rstudio.com/products/rstudio/download/preview/">RStudio Desktop Preview</a> as discussed on the <a href="https://twitter.com/LIBDrstats/status/1312187753371561984?s=20">2020-10-02 LIBD rstats club session</a>. </span></li>
</ul>
<p>Once you have installed R and RStudio, please install the following R packages:</p>
<pre class="r"><code>## For installing R packages from CRAN and GitHub
if (!requireNamespace(&quot;remotes&quot;, quietly = TRUE)) {
    install.packages(&quot;remotes&quot;)
}

## For installing Bioconductor packages
remotes::install_cran(&quot;BiocManager&quot;)

## Workflow-related R packages
remotes::install_cran(c(
    &quot;rmarkdown&quot;,
    &quot;sessioninfo&quot;,
    &quot;getopt&quot;,
    &quot;here&quot;,
    &quot;tidyr&quot;,
    &quot;usethis&quot;
))
remotes::install_github(&quot;LieberInstitute/jaffelab&quot;)

## Differential-expression related R packages
BiocManager::install(c(&quot;limma&quot;, &quot;edgeR&quot;, &quot;SummarizedExperiment&quot;, &quot;recount&quot;, &quot;ExploreModelMatrix&quot;))

## Genomics-related R packages
BiocManager::install(c(
    &quot;clusterProfiler&quot;, &quot;org.Hs.eg.db&quot;,
    &quot;VariantAnnotation&quot;
))

## Visualization-related R packages
remotes::install_cran(c(
    &quot;pheatmap&quot;,
    &quot;RColorBrewer&quot;,
    &quot;ggplot2&quot;,
    &quot;plotly&quot;
))
BiocManager::install(c(&quot;iSEE&quot;, &quot;enrichplot&quot;))

## Extra ones
remotes::install_cran(c(&quot;knitcitations&quot;, &quot;statmod&quot;))
BiocManager::install(c(&quot;BiocStyle&quot;))</code></pre>
</div>
<div id="bootcamp-plan" class="section level1" number="2">
<h1><span class="header-section-number">2</span> Bootcamp plan</h1>
<ul>
<li>Day 1: Monday October 5th 2020, 3-5 pm
<ul>
<li>RNA sequencing primer</li>
<li>LIBD’s RNA-seq processing pipeline: SPEAQeasy</li>
<li><em><a href="https://bioconductor.org/packages/3.12/SummarizedExperiment">SummarizedExperiment</a></em> overview</li>
<li>Exploring quality control metrics</li>
</ul></li>
<li>Day 2: Tuesday October 6th 2020, 3-5 pm
<ul>
<li>Exploring gene expression</li>
<li>Identifying sample swaps</li>
<li>Statistical modeling</li>
</ul></li>
<li>Day 3: Wednesday October 7th 2020, 1-3 pm
<ul>
<li>Identifying differentially expressed genes</li>
<li>Visualizing differentially expressed genes</li>
<li>Gene ontology enrichment analyses</li>
<li>Beyond SPEAQeasy</li>
</ul></li>
</ul>
</div>
<div id="day-1" class="section level1" number="3">
<h1><span class="header-section-number">3</span> Day 1</h1>
<div id="rna-sequencing-primer" class="section level2" number="3.1">
<h2><span class="header-section-number">3.1</span> RNA sequencing primer</h2>
<p><a href="https://docs.google.com/presentation/d/1X3U2lcf75Yh9BK_Scs-O3M8b9-JauUNN8ZO_xhlKs7w/edit?usp=sharing">Slides</a></p>
<p>For more details, I have some older slides on the technology for RNA-seq from <a href="http://lcolladotor.github.io/courses/PDCB-HTS.html">PDCB-HTS 2010</a>.</p>
</div>
<div id="speaqeasy-overview" class="section level2" number="3.2">
<h2><span class="header-section-number">3.2</span> SPEAQeasy overview</h2>
<p>Let’s go to the <a href="http://research.libd.org/SPEAQeasy/">SPEAQeasy</a> documentation website.</p>
<p>We’ll mostly be using the <a href="http://research.libd.org/SPEAQeasy/outputs.html">SPEAQeasy main output files</a>.</p>
<div id="speaqeasy-example" class="section level3" number="3.2.1">
<h3><span class="header-section-number">3.2.1</span> SPEAQeasy-example</h3>
<p>Example data using <a href="http://research.libd.org/SPEAQeasy-example/">SPEAQeasy-example</a> with 40 samples from the Bipolar project with Peter Zandi et al.</p>
<p>It shows:</p>
<ol style="list-style-type: decimal">
<li>How to download the example data</li>
<li>How to run SPEAQeasy with this data</li>
<li>How to identify sample swaps</li>
<li>How to run differential expression analyses with this data</li>
</ol>
<p>We’ll view these links, but we’ll start with:</p>
<ul>
<li><a href="http://research.libd.org/SPEAQeasy-example/index.html">Home</a></li>
<li><a href="http://research.libd.org/SPEAQeasy-example/prepare_data.html">Download data</a></li>
</ul>
</div>
</div>
<div id="summarizedexperiment-overview" class="section level2" number="3.3">
<h2><span class="header-section-number">3.3</span> SummarizedExperiment overview</h2>
<div id="general" class="section level3" number="3.3.1">
<h3><span class="header-section-number">3.3.1</span> General</h3>
<p>In general:</p>
<ul>
<li><em><a href="https://bioconductor.org/packages/3.12/SummarizedExperiment">SummarizedExperiment</a></em>
<ul>
<li><a href="http://bioconductor.org/packages/release/bioc/vignettes/SummarizedExperiment/inst/doc/SummarizedExperiment.html">Introduction vignette</a></li>
</ul></li>
<li>Bioconductor 2015 paper
<ul>
<li>DOI: <a href="https://doi.org/10.1038/nmeth.3252">10.1038/nmeth.3252</a></li>
</ul></li>
</ul>
</div>
<div id="example-data" class="section level3" number="3.3.2">
<h3><span class="header-section-number">3.3.2</span> Example data</h3>
<p>Let’s look at the example data</p>
<pre class="r"><code>## For loading the data
library(&quot;SummarizedExperiment&quot;)
library(&quot;here&quot;)

## Load the example data
load(here(&quot;rse_speaqeasy.RData&quot;), verbose = TRUE)</code></pre>
<pre><code>## Loading objects:
##   rse_gene</code></pre>
<p>Let’s look at the sample metadata we have in this example data. Some of these variables are defined in the <a href="http://research.libd.org/SPEAQeasy/outputs.html#quality-metrics">SPEAQeasy documentation</a> that we’ll be using shortly.</p>
<pre class="r"><code>colData(rse_gene)</code></pre>
<pre><code>## DataFrame with 40 rows and 67 columns
##                         SAMPLE_ID FQCbasicStats perBaseQual perTileQual  perSeqQual perBaseContent   GCcontent
##                       &lt;character&gt;   &lt;character&gt; &lt;character&gt; &lt;character&gt; &lt;character&gt;    &lt;character&gt; &lt;character&gt;
## R13896_H7JKMBBXX R13896_H7JKMBBXX          PASS        PASS        WARN        PASS      FAIL/WARN        FAIL
## R13903_HCTYLBBXX R13903_HCTYLBBXX          PASS        PASS   WARN/PASS        PASS      FAIL/WARN        FAIL
## R13904_H7K5NBBXX R13904_H7K5NBBXX          PASS        FAIL        FAIL        PASS      FAIL/WARN        FAIL
## R13983_H7JKMBBXX R13983_H7JKMBBXX          PASS        PASS   PASS/WARN        PASS      FAIL/WARN        FAIL
## R13985_H7JM5BBXX R13985_H7JM5BBXX          PASS        PASS        WARN        PASS      FAIL/WARN        FAIL
## ...                           ...           ...         ...         ...         ...            ...         ...
## R15068_HFFGHBBXX R15068_HFFGHBBXX          PASS        PASS        PASS        PASS      FAIL/WARN        FAIL
## R15090_HF5JNBBXX R15090_HF5JNBBXX          PASS        PASS        WARN        PASS      FAIL/WARN        FAIL
## R15093_HFY2MBBXX R15093_HFY2MBBXX          PASS        PASS        WARN        PASS      FAIL/WARN        WARN
## R15120_HFY2MBBXX R15120_HFY2MBBXX          PASS        PASS        WARN        PASS      FAIL/WARN   WARN/FAIL
## R15134_HFFGHBBXX R15134_HFFGHBBXX          PASS        PASS        PASS        PASS      FAIL/WARN   FAIL/WARN
##                     Ncontent SeqLengthDist SeqDuplication OverrepSeqs AdapterContent KmerContent SeqLength_R1
##                  &lt;character&gt;   &lt;character&gt;    &lt;character&gt; &lt;character&gt;    &lt;character&gt; &lt;character&gt;  &lt;character&gt;
## R13896_H7JKMBBXX        PASS          PASS           FAIL        FAIL           PASS          NA          101
## R13903_HCTYLBBXX        PASS          PASS           FAIL        FAIL           WARN          NA          101
## R13904_H7K5NBBXX        FAIL          PASS      PASS/WARN        WARN           PASS          NA          101
## R13983_H7JKMBBXX        PASS          PASS           FAIL        WARN           PASS          NA          101
## R13985_H7JM5BBXX        PASS          PASS           FAIL        FAIL           PASS          NA          101
## ...                      ...           ...            ...         ...            ...         ...          ...
## R15068_HFFGHBBXX        PASS          PASS           FAIL   FAIL/WARN           WARN          NA          101
## R15090_HF5JNBBXX        PASS          PASS           FAIL        WARN           WARN          NA          101
## R15093_HFY2MBBXX        PASS          PASS           FAIL        FAIL           WARN          NA          101
## R15120_HFY2MBBXX        PASS          PASS           FAIL        FAIL           WARN          NA          101
## R15134_HFFGHBBXX        PASS          PASS      FAIL/WARN        WARN           WARN          NA          101
##                  percentGC_R1 phred20-21_R1 phred48-49_R1 phred76-77_R1 phred100-101_R1 phredGT30_R1 phredGT35_R1
##                   &lt;character&gt;   &lt;character&gt;   &lt;character&gt;   &lt;character&gt;     &lt;character&gt;    &lt;numeric&gt;    &lt;numeric&gt;
## R13896_H7JKMBBXX           49          41.0          41.0          41.0            41.0           NA           NA
## R13903_HCTYLBBXX           48          41.0          41.0          41.0            41.0           NA           NA
## R13904_H7K5NBBXX           51          41.0          32.0          41.0            36.5           NA           NA
## R13983_H7JKMBBXX           48          41.0          41.0          41.0            41.0           NA           NA
## R13985_H7JM5BBXX           49          41.0          41.0          41.0            41.0           NA           NA
## ...                       ...           ...           ...           ...             ...          ...          ...
## R15068_HFFGHBBXX           50          41.0          41.0          41.0            41.0           NA           NA
## R15090_HF5JNBBXX           49          41.0          41.0          41.0            41.0           NA           NA
## R15093_HFY2MBBXX           45          41.0          41.0          41.0            41.0           NA           NA
## R15120_HFY2MBBXX           47          41.0          41.0          41.0            41.0           NA           NA
## R15134_HFFGHBBXX           47          41.0          41.0          41.0            41.0           NA           NA
##                  Adapter50-51_R1 Adapter70-71_R1 Adapter90_R1 SeqLength_R2 percentGC_R2 phred20-21_R2 phred48-49_R2
##                        &lt;numeric&gt;       &lt;numeric&gt;    &lt;numeric&gt;  &lt;character&gt;  &lt;character&gt;   &lt;character&gt;   &lt;character&gt;
## R13896_H7JKMBBXX       0.0655693        0.203478     0.828558          101           51          41.0          41.0
## R13903_HCTYLBBXX       0.1480005        1.783971     8.469852          101           49          41.0          41.0
## R13904_H7K5NBBXX       0.0374211        0.136403     0.749735          101           52          41.0          41.0
## R13983_H7JKMBBXX       0.0766332        0.223703     0.949268          101           49          41.0          41.0
## R13985_H7JM5BBXX       0.0473371        0.185952     0.963513          101           50          41.0          41.0
## ...                          ...             ...          ...          ...          ...           ...           ...
## R15068_HFFGHBBXX       0.1383892        1.491060      7.61316          101           50          41.0          41.0
## R15090_HF5JNBBXX       0.1579186        1.738622      9.43662          101           49          41.0          41.0
## R15093_HFY2MBBXX       1.2288606        2.036314      5.28201          101           46          41.0          41.0
## R15120_HFY2MBBXX       0.2441761        1.343685      5.56046          101           47          41.0          41.0
## R15134_HFFGHBBXX       0.0814902        0.947903      5.77523          101           47          41.0          41.0
##                  phred76-77_R2 phred100-101_R2 phredGT30_R2 phredGT35_R2 Adapter50-51_R2 Adapter70-71_R2 Adapter90_R2
##                    &lt;character&gt;     &lt;character&gt;    &lt;numeric&gt;    &lt;numeric&gt;       &lt;numeric&gt;       &lt;numeric&gt;    &lt;numeric&gt;
## R13896_H7JKMBBXX          41.0            41.0           NA           NA       0.0770981        0.221583     0.834020
## R13903_HCTYLBBXX          41.0            39.0           NA           NA       0.1469038        1.746648     8.258178
## R13904_H7K5NBBXX          34.5            34.5           NA           NA       0.0378473        0.123074     0.513505
## R13983_H7JKMBBXX          41.0            41.0           NA           NA       0.0892666        0.242025     0.958946
## R13985_H7JM5BBXX          41.0            41.0           NA           NA       0.0586630        0.204227     0.974275
## ...                        ...             ...          ...          ...             ...             ...          ...
## R15068_HFFGHBBXX          41.0            39.0           NA           NA       0.1379108        1.457060      7.44452
## R15090_HF5JNBBXX          41.0            41.0           NA           NA       0.1597005        1.719364      9.26811
## R15093_HFY2MBBXX          41.0            41.0           NA           NA       1.2349352        2.037324      5.25166
## R15120_HFY2MBBXX          41.0            41.0           NA           NA       0.2483485        1.340853      5.49817
## R15134_HFFGHBBXX          37.0            37.0           NA           NA       0.0786028        0.868326      5.27811
##                                 bamFile   trimmed  numReads numMapped numUnmapped overallMapRate concordMapRate
##                             &lt;character&gt; &lt;logical&gt; &lt;numeric&gt; &lt;numeric&gt;   &lt;numeric&gt;      &lt;numeric&gt;      &lt;numeric&gt;
## R13896_H7JKMBBXX ./R13896_H7JKMBBXX_a..     FALSE 106584958  99089320     7495638         0.9297         0.8862
## R13903_HCTYLBBXX ./R13903_HCTYLBBXX_a..     FALSE 134229992 111665224    22564768         0.8319         0.7780
## R13904_H7K5NBBXX ./R13904_H7K5NBBXX_a..     FALSE  61243472  40741780    20501692         0.6652         0.5198
## R13983_H7JKMBBXX ./R13983_H7JKMBBXX_a..     FALSE 144442600 134139823    10302777         0.9287         0.8839
## R13985_H7JM5BBXX ./R13985_H7JM5BBXX_a..     FALSE 130903694 121295899     9607795         0.9266         0.8901
## ...                                 ...       ...       ...       ...         ...            ...            ...
## R15068_HFFGHBBXX ./R15068_HFFGHBBXX_a..     FALSE 107872614  92780134    15092480         0.8601         0.7986
## R15090_HF5JNBBXX ./R15090_HF5JNBBXX_a..     FALSE 119870620 102489992    17380628         0.8550         0.7914
## R15093_HFY2MBBXX ./R15093_HFY2MBBXX_a..     FALSE 141870034 123395884    18474150         0.8698         0.8271
## R15120_HFY2MBBXX ./R15120_HFY2MBBXX_a..     FALSE 106271242  90596496    15674746         0.8525         0.8023
## R15134_HFFGHBBXX ./R15134_HFFGHBBXX_a..     FALSE 120451238 101822895    18628343         0.8453         0.7509
##                  totalMapped mitoMapped  mitoRate totalAssignedGene gene_Assigned gene_Unassigned_Unmapped
##                    &lt;numeric&gt;  &lt;numeric&gt; &lt;numeric&gt;         &lt;numeric&gt;     &lt;integer&gt;                &lt;integer&gt;
## R13896_H7JKMBBXX   160944960    2750646 0.0168034          0.256746      21573488                        0
## R13903_HCTYLBBXX   139882024    2825252 0.0197975          0.379945      27896691                        0
## R13904_H7K5NBBXX    54288854    1326211 0.0238463          0.371944      12380935                        0
## R13983_H7JKMBBXX   196453977    3972690 0.0198212          0.288361      29704536                        0
## R13985_H7JM5BBXX   187601620    5307450 0.0275127          0.276588      27270855                        0
## ...                      ...        ...       ...               ...           ...                      ...
## R15068_HFFGHBBXX   129937217    2236204 0.0169187          0.308115      21013286                        0
## R15090_HF5JNBBXX   151483286    2642744 0.0171466          0.248909      19700029                        0
## R15093_HFY2MBBXX   151106303    4022581 0.0259306          0.373461      29720302                        0
## R15120_HFY2MBBXX   131550971    3668547 0.0271303          0.291463      20242616                        0
## R15134_HFFGHBBXX   123436361    1922519 0.0153361          0.390820      26348889                        0
##                  gene_Unassigned_Read_Type gene_Unassigned_Singleton gene_Unassigned_MappingQuality
##                                  &lt;integer&gt;                 &lt;integer&gt;                      &lt;integer&gt;
## R13896_H7JKMBBXX                         0                         0                              0
## R13903_HCTYLBBXX                         0                         0                              0
## R13904_H7K5NBBXX                         0                         0                              0
## R13983_H7JKMBBXX                         0                         0                              0
## R13985_H7JM5BBXX                         0                         0                              0
## ...                                    ...                       ...                            ...
## R15068_HFFGHBBXX                         0                         0                              0
## R15090_HF5JNBBXX                         0                         0                              0
## R15093_HFY2MBBXX                         0                         0                              0
## R15120_HFY2MBBXX                         0                         0                              0
## R15134_HFFGHBBXX                         0                         0                              0
##                  gene_Unassigned_Chimera gene_Unassigned_FragmentLength gene_Unassigned_Duplicate
##                                &lt;integer&gt;                      &lt;integer&gt;                 &lt;integer&gt;
## R13896_H7JKMBBXX                       0                              0                         0
## R13903_HCTYLBBXX                       0                              0                         0
## R13904_H7K5NBBXX                       0                              0                         0
## R13983_H7JKMBBXX                       0                              0                         0
## R13985_H7JM5BBXX                       0                              0                         0
## ...                                  ...                            ...                       ...
## R15068_HFFGHBBXX                       0                              0                         0
## R15090_HF5JNBBXX                       0                              0                         0
## R15093_HFY2MBBXX                       0                              0                         0
## R15120_HFY2MBBXX                       0                              0                         0
## R15134_HFFGHBBXX                       0                              0                         0
##                  gene_Unassigned_MultiMapping gene_Unassigned_Secondary gene_Unassigned_NonSplit
##                                     &lt;integer&gt;                 &lt;integer&gt;                &lt;integer&gt;
## R13896_H7JKMBBXX                     43753848                         0                        0
## R13903_HCTYLBBXX                     23142526                         0                        0
## R13904_H7K5NBBXX                     13059390                         0                        0
## R13983_H7JKMBBXX                     45549411                         0                        0
## R13985_H7JM5BBXX                     48981261                         0                        0
## ...                                       ...                       ...                      ...
## R15068_HFFGHBBXX                     28766178                         0                        0
## R15090_HF5JNBBXX                     35626413                         0                        0
## R15093_HFY2MBBXX                     22349534                         0                        0
## R15120_HFY2MBBXX                     30546708                         0                        0
## R15134_HFFGHBBXX                     17710168                         0                        0
##                  gene_Unassigned_NoFeatures gene_Unassigned_Overlapping_Length gene_Unassigned_Ambiguity   rRNA_rate
##                                   &lt;integer&gt;                          &lt;integer&gt;                 &lt;integer&gt;   &lt;numeric&gt;
## R13896_H7JKMBBXX                   14465278                                  0                   4233906 4.35720e-06
## R13903_HCTYLBBXX                   16347297                                  0                   6036512 1.72063e-06
## R13904_H7K5NBBXX                    4073327                                  0                   3773441 2.31000e-05
## R13983_H7JKMBBXX                   21412703                                  0                   6345085 2.89518e-06
## R13985_H7JM5BBXX                   14890196                                  0                   7455151 8.39724e-06
## ...                                     ...                                ...                       ...         ...
## R15068_HFFGHBBXX                   12351911                                  0                   6068022 4.04506e-06
## R15090_HF5JNBBXX                   20705757                                  0                   3113252 5.06091e-05
## R15093_HFY2MBBXX                   24290093                                  0                   3220748 1.04306e-06
## R15120_HFY2MBBXX                   16532910                                  0                   2129438 4.94007e-06
## R15134_HFFGHBBXX                   19654182                                  0                   3706256 7.09707e-06
##                  BrainRegion     Race PrimaryDx      Sex  AgeDeath       BrNum
##                     &lt;factor&gt; &lt;factor&gt;  &lt;factor&gt; &lt;factor&gt; &lt;numeric&gt; &lt;character&gt;
## R13896_H7JKMBBXX    Amygdala     CAUC   Bipolar        M     60.54      Br1992
## R13903_HCTYLBBXX    Amygdala     CAUC   Bipolar        M     45.46      Br1677
## R13904_H7K5NBBXX    Amygdala     CAUC   Bipolar        M     26.18      Br1734
## R13983_H7JKMBBXX    Amygdala     CAUC   Bipolar        F     44.35      Br2262
## R13985_H7JM5BBXX    Amygdala     CAUC   Control        F     48.99      Br1753
## ...                      ...      ...       ...      ...       ...         ...
## R15068_HFFGHBBXX    sACC         CAUC   Bipolar        F   36.6022      Br5867
## R15090_HF5JNBBXX    Amygdala     CAUC   Bipolar        M   32.3642      Br5979
## R15093_HFY2MBBXX    Amygdala     CAUC   Bipolar        F   33.0897      Br6016
## R15120_HFY2MBBXX    Amygdala     CAUC   Control        F   71.0600      Br1756
## R15134_HFFGHBBXX    sACC         CAUC   Control        F   49.0267      Br5931</code></pre>
<p>SPEAQeasy can align data to the human or mouse genomes using specific Gencode annotations. The example data we are working with is from human using Gencode v32.</p>
<pre class="r"><code>rowData(rse_gene)</code></pre>
<pre><code>## DataFrame with 60609 rows and 10 columns
##                      Length         gencodeID       ensemblID              gene_type      Symbol  EntrezID       Class
##                   &lt;integer&gt;       &lt;character&gt;     &lt;character&gt;            &lt;character&gt; &lt;character&gt; &lt;integer&gt; &lt;character&gt;
## ENSG00000223972.5      1735 ENSG00000223972.5 ENSG00000223972 transcribed_unproces..     DDX11L1        NA       InGen
## ENSG00000227232.5      1351 ENSG00000227232.5 ENSG00000227232 unprocessed_pseudogene      WASH7P        NA       InGen
## ENSG00000278267.1        68 ENSG00000278267.1 ENSG00000278267                  miRNA   MIR6859-1        NA       InGen
## ENSG00000243485.5      1021 ENSG00000243485.5 ENSG00000243485                 lncRNA MIR1302-2HG        NA       InGen
## ENSG00000284332.1       138 ENSG00000284332.1 ENSG00000284332                  miRNA   MIR1302-2        NA       InGen
## ...                     ...               ...             ...                    ...         ...       ...         ...
## ENSG00000198695.2       525 ENSG00000198695.2 ENSG00000198695         protein_coding      MT-ND6      4541       InGen
## ENSG00000210194.1        69 ENSG00000210194.1 ENSG00000210194                Mt_tRNA       MT-TE        NA       InGen
## ENSG00000198727.2      1141 ENSG00000198727.2 ENSG00000198727         protein_coding      MT-CYB      4519       InGen
## ENSG00000210195.2        66 ENSG00000210195.2 ENSG00000210195                Mt_tRNA       MT-TT        NA       InGen
## ENSG00000210196.2        68 ENSG00000210196.2 ENSG00000210196                Mt_tRNA       MT-TP        NA       InGen
##                    meanExprs     NumTx              gencodeTx
##                    &lt;numeric&gt; &lt;integer&gt;            &lt;character&gt;
## ENSG00000223972.5 0.00347263         2 ENST00000456328.2;EN..
## ENSG00000227232.5 1.57167877         1      ENST00000488147.1
## ENSG00000278267.1 7.74437941         1      ENST00000619216.1
## ENSG00000243485.5 0.01403484         2 ENST00000473358.1;EN..
## ENSG00000284332.1 0.00000000         1      ENST00000607096.1
## ...                      ...       ...                    ...
## ENSG00000198695.2  182.92863         1      ENST00000361681.2
## ENSG00000210194.1    9.80684         1      ENST00000387459.1
## ENSG00000198727.2  826.87424         1      ENST00000361789.2
## ENSG00000210195.2  276.37225         1      ENST00000387460.2
## ENSG00000210196.2  366.58138         1      ENST00000387461.2</code></pre>
<p>The bulk of the data is stored in the gene count matrix generated using <code>featureCounts</code> by <code>SPEAQeasy</code>. This is a large matrix with one row per gene and one column per sample. As such, we’ll just explore a tiny piece of it.</p>
<pre class="r"><code>## The names of the count matrices we have
assayNames(rse_gene)</code></pre>
<pre><code>## [1] &quot;counts&quot;</code></pre>
<pre class="r"><code>## The full &quot;counts&quot; matrix
dim(assay(rse_gene, &quot;counts&quot;))</code></pre>
<pre><code>## [1] 60609    40</code></pre>
<pre class="r"><code>## The top left corner of the matrix counts
jaffelab::corner(assay(rse_gene, &quot;counts&quot;))</code></pre>
<pre><code>##                   R13896_H7JKMBBXX R13903_HCTYLBBXX R13904_H7K5NBBXX R13983_H7JKMBBXX R13985_H7JM5BBXX R13986_H7JM5BBXX
## ENSG00000223972.5                0                0                0                0                0                0
## ENSG00000227232.5               46               36               20              118               61               47
## ENSG00000278267.1                8                3                0               13                4                8
## ENSG00000243485.5                0                0                0                0                1                0
## ENSG00000284332.1                0                0                0                0                0                0
## ENSG00000237613.2                0                0                0                0                0                0</code></pre>
</div>
</div>
<div id="exploring-quality-control-metrics" class="section level2" number="3.4">
<h2><span class="header-section-number">3.4</span> Exploring quality control metrics</h2>
<p>The <a href="http://research.libd.org/SPEAQeasy/outputs.html#quality-metrics">SPEAQeasy documentation</a> explains the meaning of some of the quality metrics that are included in the RSE files generated by this processing pipeline. Let’s explore these metrics and their relationship to some of our sample metadata we manually included, such as the age at time of death and primary diagnosis.</p>
<p>We can start with some boxplots comparing some metrics against primary diagnosis and brain region, as shown in the <a href="http://research.libd.org/SPEAQeasy-example/de_analysis_speaqeasy.html#14_Stats_vs_diagnosis_and_brain_region">SPEAQeasy-example differential expression script</a>.</p>
<p>We can also use tools like <em><a href="https://CRAN.R-project.org/package=ggplot2">ggplot2</a></em> and <em><a href="https://CRAN.R-project.org/package=plotly">plotly</a></em> to interactively explore some metrics. For doing so, it’ll be useful to check some of our LIBD rstats club sessions on:</p>
<ul>
<li><a href="https://twitter.com/LIBDrstats/status/1308934265267195905?s=20">High quality graphics in R</a></li>
<li><a href="https://twitter.com/LIBDrstats/status/1304799631386308608?s=20">Interactive graphics with plotly</a></li>
</ul>
<div id="exercise" class="section level3" number="3.4.1">
<h3><span class="header-section-number">3.4.1</span> Exercise</h3>
<p>Adapt the interactive graphics code from the LIBD rstats club session available <a href="https://gist.github.com/lcolladotor/350ac8843e153d135f64880a1d029b14">here</a> for our data.</p>
</div>
<div id="exercise-hints" class="section level3" number="3.4.2">
<h3><span class="header-section-number">3.4.2</span> Exercise hints</h3>
<pre class="r"><code>## First we load the required packages
library(&quot;ggplot2&quot;)
library(&quot;plotly&quot;)

## Next, we convert the sample metadata to a data.frame
## which will make it easier to work with
pd_df &lt;- as.data.frame(colData(rse_gene))

## We need a &quot;key&quot; variable that is unique

## Let&#39;s make the key

## Let&#39;s make a &quot;highlighted&quot; table

## Make a plot using the highlight table

## Make a second plot

## Convert them to interactive plots

## Now group them together

## Fancy one</code></pre>
</div>
<div id="exercise-solution" class="section level3" number="3.4.3">
<h3><span class="header-section-number">3.4.3</span> Exercise solution</h3>
<div id="step-1" class="section level4" number="3.4.3.1">
<h4><span class="header-section-number">3.4.3.1</span> Step 1</h4>
<p>As a first step, let’s add the code from the gist that is listed below each of the comments from the previous <a href="https://gist.github.com/lcolladotor/350ac8843e153d135f64880a1d029b14">code</a>.</p>
<pre class="r"><code>## First we load the required packages
library(&quot;ggplot2&quot;)
library(&quot;plotly&quot;)

## Next, we convert the sample metadata to a data.frame
## which will make it easier to work with
pd_df &lt;- as.data.frame(colData(rse_gene))

## We need a &quot;key&quot; variable that is unique
length(unique(pd_df$RNum))
nrow(pd_df)

## Let&#39;s make the key
pd_df$key &lt;- pd_df$RNum

## Let&#39;s make a &quot;highlighted&quot; table
pd_key &lt;- highlight_key(pd_df, ~key)

## Make a plot using the highlight table
gg_mean_mito_vs_mean_gene &lt;-
    ggplot(
        pd_key,
        aes(x = mean_mitoRate, y = mean_totalAssignedGene, color = Region)
    ) +
    geom_point()
gg_mean_mito_vs_mean_gene

## Make a second plot
gg_mean_mito_vs_mean_RIN &lt;-
    ggplot(pd_key, aes(x = mean_mitoRate, y = mean_RIN, color = Region)) +
    geom_point()
gg_mean_mito_vs_mean_RIN

## Convert them to interactive plots
p_mean_mito_vs_mean_gene &lt;- ggplotly(gg_mean_mito_vs_mean_gene)
p_mean_mito_vs_mean_RIN &lt;- ggplotly(gg_mean_mito_vs_mean_RIN)

## Now group them together
p_merged &lt;- subplot(
    p_mean_mito_vs_mean_gene,
    p_mean_mito_vs_mean_RIN,
    nrows = 1,
    shareX = TRUE,
    shareY = FALSE,
    which_layout = 2
)

## Fancy one
p_fancy &lt;- highlight(
    p_merged,
    on = &quot;plotly_click&quot;,
    selectize = TRUE,
    dynamic = TRUE,
    persistent = TRUE
)</code></pre>
</div>
<div id="step-2" class="section level4" number="3.4.3.2">
<h4><span class="header-section-number">3.4.3.2</span> Step 2</h4>
<p>Now that we have the code, we need to adapt it. Some variable names are slightly different, some don’t exist. This process will typically involve several error messages while we figure out what’s there and what’s different. The end result will look like the following code.</p>
<pre class="r"><code>## First we load the required packages
library(&quot;ggplot2&quot;)
library(&quot;plotly&quot;)

## Next, we convert the sample metadata to a data.frame
## which will make it easier to work with
pd_df &lt;- as.data.frame(colData(rse_gene))

## We need a &quot;key&quot; variable that is unique
## We replaced RNum by SAMPLE_ID
length(unique(pd_df$SAMPLE_ID))</code></pre>
<pre><code>## [1] 40</code></pre>
<pre class="r"><code>nrow(pd_df)</code></pre>
<pre><code>## [1] 40</code></pre>
<pre class="r"><code>## Let&#39;s make the key
pd_df$key &lt;- pd_df$SAMPLE_ID

## Let&#39;s make a &quot;highlighted&quot; table
pd_key &lt;- highlight_key(pd_df, ~key)

## Make a plot using the highlight table
## Here we remoted &quot;mean_&quot; from the variable and object names
## and we changed Region to BrainRegion
gg_mito_vs_gene &lt;-
    ggplot(
        pd_key,
        aes(x = mitoRate, y = totalAssignedGene, color = BrainRegion)
    ) +
    geom_point()

## Make a second plot
## We don&#39;t have RIN, so we&#39;ll use rRNA_rate here
gg_mito_vs_rRNA_rate &lt;-
    ggplot(pd_key, aes(x = mitoRate, y = rRNA_rate, color = BrainRegion)) +
    geom_point()

## Convert them to interactive plots
p_mito_vs_gene &lt;- ggplotly(gg_mito_vs_gene)
p_mito_vs_rRNA_rate &lt;- ggplotly(gg_mito_vs_rRNA_rate)

## Now group them together
p_merged &lt;- subplot(
    p_mito_vs_gene,
    p_mito_vs_rRNA_rate,
    nrows = 1,
    shareX = TRUE,
    shareY = FALSE,
    which_layout = 2
)

## Fancy one
p_fancy &lt;- highlight(
    p_merged,
    on = &quot;plotly_click&quot;,
    selectize = TRUE,
    dynamic = TRUE,
    persistent = TRUE
)</code></pre>
<pre><code>## Adding more colors to the selection color palette.</code></pre>
<pre><code>## We recommend setting `persistent` to `FALSE` (the default) because persistent selection mode can now be used by holding the shift key (while triggering the `on` event).</code></pre>
<pre class="r"><code>## End result
p_fancy</code></pre>
<pre><code>## Setting the `off` event (i.e., &#39;plotly_doubleclick&#39;) to match the `on` event (i.e., &#39;plotly_click&#39;). You can change this default via the `highlight()` function.</code></pre>
<div id="htmlwidget-cee46415146e7d618930" style="width:100%;height:480px;" class="plotly html-widget"></div>
<script type="application/json" data-for="htmlwidget-cee46415146e7d618930">{"x":{"data":[{"x":[0.0168034198792117,0.0197975329583055,0.0238462546074521,0.0198211648153586,0.0275127032648076,0.0288706308372253,0.0234725936031733,0.0213304369420574,0.021964661021335,0.0155062551330653,0.0207495317198981,0.017439187179823,0.0303682303401419,0.0147083478989621,0.0130221144878825,0.0177429776793537,0.016246363929729,0.017146642912946,0.0259305739606816,0.027130306735748],"y":[0.256746179658517,0.379944719249245,0.371944014456294,0.288360699875602,0.276587796178894,0.250509358179923,0.348872446684914,0.138408226770101,0.392097545014564,0.368049437343863,0.39016474568347,0.233746263186425,0.226913163437878,0.215303934788801,0.329750338532713,0.215822176515167,0.0952945195182315,0.248909176094025,0.373461286337134,0.29146333582869],"text":[".crossTalkKey: R13896_H7JKMBBXX<br />mitoRate: 0.01680342<br />totalAssignedGene: 0.25674618<br />BrainRegion: Amygdala",".crossTalkKey: R13903_HCTYLBBXX<br />mitoRate: 0.01979753<br />totalAssignedGene: 0.37994472<br />BrainRegion: Amygdala",".crossTalkKey: R13904_H7K5NBBXX<br />mitoRate: 0.02384625<br />totalAssignedGene: 0.37194401<br />BrainRegion: Amygdala",".crossTalkKey: R13983_H7JKMBBXX<br />mitoRate: 0.01982116<br />totalAssignedGene: 0.28836070<br />BrainRegion: Amygdala",".crossTalkKey: R13985_H7JM5BBXX<br />mitoRate: 0.02751270<br />totalAssignedGene: 0.27658780<br />BrainRegion: Amygdala",".crossTalkKey: R13986_H7JM5BBXX<br />mitoRate: 0.02887063<br />totalAssignedGene: 0.25050936<br />BrainRegion: Amygdala",".crossTalkKey: R13995_H7JHNBBXX<br />mitoRate: 0.02347259<br />totalAssignedGene: 0.34887245<br />BrainRegion: Amygdala",".crossTalkKey: R14017_HF5JNBBXX<br />mitoRate: 0.02133044<br />totalAssignedGene: 0.13840823<br />BrainRegion: Amygdala",".crossTalkKey: R14023_H7JHNBBXX<br />mitoRate: 0.02196466<br />totalAssignedGene: 0.39209755<br />BrainRegion: Amygdala",".crossTalkKey: R14030_H7K5NBBXX<br />mitoRate: 0.01550626<br />totalAssignedGene: 0.36804944<br />BrainRegion: Amygdala",".crossTalkKey: R14031_HF5JNBBXX<br />mitoRate: 0.02074953<br />totalAssignedGene: 0.39016475<br />BrainRegion: Amygdala",".crossTalkKey: R14039_H7K5NBBXX<br />mitoRate: 0.01743919<br />totalAssignedGene: 0.23374626<br />BrainRegion: Amygdala",".crossTalkKey: R14069_HCTYLBBXX<br />mitoRate: 0.03036823<br />totalAssignedGene: 0.22691316<br />BrainRegion: Amygdala",".crossTalkKey: R14071_HF3JYBBXX<br />mitoRate: 0.01470835<br />totalAssignedGene: 0.21530393<br />BrainRegion: Amygdala",".crossTalkKey: R14073_HF3JYBBXX<br />mitoRate: 0.01302211<br />totalAssignedGene: 0.32975034<br />BrainRegion: Amygdala",".crossTalkKey: R14077_HCTYLBBXX<br />mitoRate: 0.01774298<br />totalAssignedGene: 0.21582218<br />BrainRegion: Amygdala",".crossTalkKey: R15058_HF5JNBBXX<br />mitoRate: 0.01624636<br />totalAssignedGene: 0.09529452<br />BrainRegion: Amygdala",".crossTalkKey: R15090_HF5JNBBXX<br />mitoRate: 0.01714664<br />totalAssignedGene: 0.24890918<br />BrainRegion: Amygdala",".crossTalkKey: R15093_HFY2MBBXX<br />mitoRate: 0.02593057<br />totalAssignedGene: 0.37346129<br />BrainRegion: Amygdala",".crossTalkKey: R15120_HFY2MBBXX<br />mitoRate: 0.02713031<br />totalAssignedGene: 0.29146334<br />BrainRegion: Amygdala"],"key":["R13896_H7JKMBBXX","R13903_HCTYLBBXX","R13904_H7K5NBBXX","R13983_H7JKMBBXX","R13985_H7JM5BBXX","R13986_H7JM5BBXX","R13995_H7JHNBBXX","R14017_HF5JNBBXX","R14023_H7JHNBBXX","R14030_H7K5NBBXX","R14031_HF5JNBBXX","R14039_H7K5NBBXX","R14069_HCTYLBBXX","R14071_HF3JYBBXX","R14073_HF3JYBBXX","R14077_HCTYLBBXX","R15058_HF5JNBBXX","R15090_HF5JNBBXX","R15093_HFY2MBBXX","R15120_HFY2MBBXX"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(248,118,109,1)","opacity":1,"size":5.66929133858268,"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(248,118,109,1)"}},"hoveron":"points","set":"SharedData76ab4600","name":"Amygdala","legendgroup":"Amygdala","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","_isNestedKey":false,"frame":null},{"x":[0.018266623830598,0.0156842807997245,0.0179850683285263,0.014226259970725,0.0158331315618843,0.0273801375999526,0.0301852114180031,0.0145754167453164,0.0176381180694062,0.0186381122058921,0.0199499672846804,0.0139567542891957,0.0119439216630516,0.0199263395379048,0.0129711581306921,0.0175490352254017,0.0168091081445393,0.0140683200285263,0.0169187116674539,0.0153361213820672],"y":[0.259869683713983,0.268836831436762,0.28757848590218,0.29825925446411,0.159777378879144,0.295439771421986,0.353335796303596,0.344382754254371,0.231412531692872,0.383575682557374,0.407288206024691,0.340527326218284,0.280077256371953,0.290347891582152,0.2972027801218,0.331479322158735,0.319130262260883,0.201979179490042,0.308115422193542,0.390820029132523],"text":[".crossTalkKey: R14091_HFY2MBBXX<br />mitoRate: 0.01826662<br />totalAssignedGene: 0.25986968<br />BrainRegion: sACC",".crossTalkKey: R14102_H7K5NBBXX<br />mitoRate: 0.01568428<br />totalAssignedGene: 0.26883683<br />BrainRegion: sACC",".crossTalkKey: R14125_H7JM5BBXX<br />mitoRate: 0.01798507<br />totalAssignedGene: 0.28757849<br />BrainRegion: sACC",".crossTalkKey: R14129_HCTYLBBXX<br />mitoRate: 0.01422626<br />totalAssignedGene: 0.29825925<br />BrainRegion: sACC",".crossTalkKey: R14136_H7K5NBBXX<br />mitoRate: 0.01583313<br />totalAssignedGene: 0.15977738<br />BrainRegion: sACC",".crossTalkKey: R14156_H7JHNBBXX<br />mitoRate: 0.02738014<br />totalAssignedGene: 0.29543977<br />BrainRegion: sACC",".crossTalkKey: R14184_H7K5NBBXX<br />mitoRate: 0.03018521<br />totalAssignedGene: 0.35333580<br />BrainRegion: sACC",".crossTalkKey: R14202_H7K5NBBXX<br />mitoRate: 0.01457542<br />totalAssignedGene: 0.34438275<br />BrainRegion: sACC",".crossTalkKey: R14223_HF5JNBBXX<br />mitoRate: 0.01763812<br />totalAssignedGene: 0.23141253<br />BrainRegion: sACC",".crossTalkKey: R14225_HF3JYBBXX<br />mitoRate: 0.01863811<br />totalAssignedGene: 0.38357568<br />BrainRegion: sACC",".crossTalkKey: R14238_HF5JNBBXX<br />mitoRate: 0.01994997<br />totalAssignedGene: 0.40728821<br />BrainRegion: sACC",".crossTalkKey: R14247_HF3JYBBXX<br />mitoRate: 0.01395675<br />totalAssignedGene: 0.34052733<br />BrainRegion: sACC",".crossTalkKey: R14279_HFY2MBBXX<br />mitoRate: 0.01194392<br />totalAssignedGene: 0.28007726<br />BrainRegion: sACC",".crossTalkKey: R14290_H7L3FBBXX<br />mitoRate: 0.01992634<br />totalAssignedGene: 0.29034789<br />BrainRegion: sACC",".crossTalkKey: R14296_H7JLCBBXX<br />mitoRate: 0.01297116<br />totalAssignedGene: 0.29720278<br />BrainRegion: sACC",".crossTalkKey: R14304_H7JKMBBXX<br />mitoRate: 0.01754904<br />totalAssignedGene: 0.33147932<br />BrainRegion: sACC",".crossTalkKey: R14995_HFFGHBBXX<br />mitoRate: 0.01680911<br />totalAssignedGene: 0.31913026<br />BrainRegion: sACC",".crossTalkKey: R15016_HF5JNBBXX<br />mitoRate: 0.01406832<br />totalAssignedGene: 0.20197918<br />BrainRegion: sACC",".crossTalkKey: R15068_HFFGHBBXX<br />mitoRate: 0.01691871<br />totalAssignedGene: 0.30811542<br />BrainRegion: sACC",".crossTalkKey: R15134_HFFGHBBXX<br />mitoRate: 0.01533612<br />totalAssignedGene: 0.39082003<br />BrainRegion: sACC"],"key":["R14091_HFY2MBBXX","R14102_H7K5NBBXX","R14125_H7JM5BBXX","R14129_HCTYLBBXX","R14136_H7K5NBBXX","R14156_H7JHNBBXX","R14184_H7K5NBBXX","R14202_H7K5NBBXX","R14223_HF5JNBBXX","R14225_HF3JYBBXX","R14238_HF5JNBBXX","R14247_HF3JYBBXX","R14279_HFY2MBBXX","R14290_H7L3FBBXX","R14296_H7JLCBBXX","R14304_H7JKMBBXX","R14995_HFFGHBBXX","R15016_HF5JNBBXX","R15068_HFFGHBBXX","R15134_HFFGHBBXX"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(0,191,196,1)","opacity":1,"size":5.66929133858268,"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(0,191,196,1)"}},"hoveron":"points","set":"SharedData76ab4600","name":"sACC","legendgroup":"sACC","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","_isNestedKey":false,"frame":null},{"x":[0.0168034198792117,0.0197975329583055,0.0238462546074521,0.0198211648153586,0.0275127032648076,0.0288706308372253,0.0234725936031733,0.0213304369420574,0.021964661021335,0.0155062551330653,0.0207495317198981,0.017439187179823,0.0303682303401419,0.0147083478989621,0.0130221144878825,0.0177429776793537,0.016246363929729,0.017146642912946,0.0259305739606816,0.027130306735748],"y":[4.35719991129853e-06,1.72063417844073e-06,2.31000324288917e-05,2.89518072256709e-06,8.39724313740805e-06,7.04055411977144e-07,3.48269955600556e-06,0.00173657681257698,1.38433657333872e-06,1.56958479763719e-05,3.16977216289404e-06,4.28664516665614e-06,0.000162093175517004,1.76877535921915e-05,2.02725638975829e-06,1.93302479922641e-05,0.000107076228446015,5.06090625551871e-05,1.04305804160402e-06,4.94007296290163e-06],"text":[".crossTalkKey: R13896_H7JKMBBXX<br />mitoRate: 0.01680342<br />rRNA_rate: 4.357200e-06<br />BrainRegion: Amygdala",".crossTalkKey: R13903_HCTYLBBXX<br />mitoRate: 0.01979753<br />rRNA_rate: 1.720634e-06<br />BrainRegion: Amygdala",".crossTalkKey: R13904_H7K5NBBXX<br />mitoRate: 0.02384625<br />rRNA_rate: 2.310003e-05<br />BrainRegion: Amygdala",".crossTalkKey: R13983_H7JKMBBXX<br />mitoRate: 0.01982116<br />rRNA_rate: 2.895181e-06<br />BrainRegion: Amygdala",".crossTalkKey: R13985_H7JM5BBXX<br />mitoRate: 0.02751270<br />rRNA_rate: 8.397243e-06<br />BrainRegion: Amygdala",".crossTalkKey: R13986_H7JM5BBXX<br />mitoRate: 0.02887063<br />rRNA_rate: 7.040554e-07<br />BrainRegion: Amygdala",".crossTalkKey: R13995_H7JHNBBXX<br />mitoRate: 0.02347259<br />rRNA_rate: 3.482700e-06<br />BrainRegion: Amygdala",".crossTalkKey: R14017_HF5JNBBXX<br />mitoRate: 0.02133044<br />rRNA_rate: 1.736577e-03<br />BrainRegion: Amygdala",".crossTalkKey: R14023_H7JHNBBXX<br />mitoRate: 0.02196466<br />rRNA_rate: 1.384337e-06<br />BrainRegion: Amygdala",".crossTalkKey: R14030_H7K5NBBXX<br />mitoRate: 0.01550626<br />rRNA_rate: 1.569585e-05<br />BrainRegion: Amygdala",".crossTalkKey: R14031_HF5JNBBXX<br />mitoRate: 0.02074953<br />rRNA_rate: 3.169772e-06<br />BrainRegion: Amygdala",".crossTalkKey: R14039_H7K5NBBXX<br />mitoRate: 0.01743919<br />rRNA_rate: 4.286645e-06<br />BrainRegion: Amygdala",".crossTalkKey: R14069_HCTYLBBXX<br />mitoRate: 0.03036823<br />rRNA_rate: 1.620932e-04<br />BrainRegion: Amygdala",".crossTalkKey: R14071_HF3JYBBXX<br />mitoRate: 0.01470835<br />rRNA_rate: 1.768775e-05<br />BrainRegion: Amygdala",".crossTalkKey: R14073_HF3JYBBXX<br />mitoRate: 0.01302211<br />rRNA_rate: 2.027256e-06<br />BrainRegion: Amygdala",".crossTalkKey: R14077_HCTYLBBXX<br />mitoRate: 0.01774298<br />rRNA_rate: 1.933025e-05<br />BrainRegion: Amygdala",".crossTalkKey: R15058_HF5JNBBXX<br />mitoRate: 0.01624636<br />rRNA_rate: 1.070762e-04<br />BrainRegion: Amygdala",".crossTalkKey: R15090_HF5JNBBXX<br />mitoRate: 0.01714664<br />rRNA_rate: 5.060906e-05<br />BrainRegion: Amygdala",".crossTalkKey: R15093_HFY2MBBXX<br />mitoRate: 0.02593057<br />rRNA_rate: 1.043058e-06<br />BrainRegion: Amygdala",".crossTalkKey: R15120_HFY2MBBXX<br />mitoRate: 0.02713031<br />rRNA_rate: 4.940073e-06<br />BrainRegion: Amygdala"],"key":["R13896_H7JKMBBXX","R13903_HCTYLBBXX","R13904_H7K5NBBXX","R13983_H7JKMBBXX","R13985_H7JM5BBXX","R13986_H7JM5BBXX","R13995_H7JHNBBXX","R14017_HF5JNBBXX","R14023_H7JHNBBXX","R14030_H7K5NBBXX","R14031_HF5JNBBXX","R14039_H7K5NBBXX","R14069_HCTYLBBXX","R14071_HF3JYBBXX","R14073_HF3JYBBXX","R14077_HCTYLBBXX","R15058_HF5JNBBXX","R15090_HF5JNBBXX","R15093_HFY2MBBXX","R15120_HFY2MBBXX"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(248,118,109,1)","opacity":1,"size":5.66929133858268,"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(248,118,109,1)"}},"hoveron":"points","set":"SharedData76ab4600","name":"Amygdala","legendgroup":"Amygdala","showlegend":true,"xaxis":"x2","yaxis":"y2","hoverinfo":"text","_isNestedKey":false,"frame":null},{"x":[0.018266623830598,0.0156842807997245,0.0179850683285263,0.014226259970725,0.0158331315618843,0.0273801375999526,0.0301852114180031,0.0145754167453164,0.0176381180694062,0.0186381122058921,0.0199499672846804,0.0139567542891957,0.0119439216630516,0.0199263395379048,0.0129711581306921,0.0175490352254017,0.0168091081445393,0.0140683200285263,0.0169187116674539,0.0153361213820672],"y":[7.91099142840564e-05,1.06936257803654e-05,7.93720086674233e-06,8.50215343014421e-06,0.000101649369184167,4.62012730573489e-06,9.19106705312332e-05,2.29096004867665e-06,4.68985849944214e-05,1.43147064351933e-06,4.05465846237076e-06,2.23909190159534e-06,9.12784552113959e-06,1.51711515962152e-05,5.13660781447827e-06,5.68272504420546e-06,0.000421975458193715,3.75338817853739e-05,4.04505987307268e-06,7.09707342878859e-06],"text":[".crossTalkKey: R14091_HFY2MBBXX<br />mitoRate: 0.01826662<br />rRNA_rate: 7.910991e-05<br />BrainRegion: sACC",".crossTalkKey: R14102_H7K5NBBXX<br />mitoRate: 0.01568428<br />rRNA_rate: 1.069363e-05<br />BrainRegion: sACC",".crossTalkKey: R14125_H7JM5BBXX<br />mitoRate: 0.01798507<br />rRNA_rate: 7.937201e-06<br />BrainRegion: sACC",".crossTalkKey: R14129_HCTYLBBXX<br />mitoRate: 0.01422626<br />rRNA_rate: 8.502153e-06<br />BrainRegion: sACC",".crossTalkKey: R14136_H7K5NBBXX<br />mitoRate: 0.01583313<br />rRNA_rate: 1.016494e-04<br />BrainRegion: sACC",".crossTalkKey: R14156_H7JHNBBXX<br />mitoRate: 0.02738014<br />rRNA_rate: 4.620127e-06<br />BrainRegion: sACC",".crossTalkKey: R14184_H7K5NBBXX<br />mitoRate: 0.03018521<br />rRNA_rate: 9.191067e-05<br />BrainRegion: sACC",".crossTalkKey: R14202_H7K5NBBXX<br />mitoRate: 0.01457542<br />rRNA_rate: 2.290960e-06<br />BrainRegion: sACC",".crossTalkKey: R14223_HF5JNBBXX<br />mitoRate: 0.01763812<br />rRNA_rate: 4.689858e-05<br />BrainRegion: sACC",".crossTalkKey: R14225_HF3JYBBXX<br />mitoRate: 0.01863811<br />rRNA_rate: 1.431471e-06<br />BrainRegion: sACC",".crossTalkKey: R14238_HF5JNBBXX<br />mitoRate: 0.01994997<br />rRNA_rate: 4.054658e-06<br />BrainRegion: sACC",".crossTalkKey: R14247_HF3JYBBXX<br />mitoRate: 0.01395675<br />rRNA_rate: 2.239092e-06<br />BrainRegion: sACC",".crossTalkKey: R14279_HFY2MBBXX<br />mitoRate: 0.01194392<br />rRNA_rate: 9.127846e-06<br />BrainRegion: sACC",".crossTalkKey: R14290_H7L3FBBXX<br />mitoRate: 0.01992634<br />rRNA_rate: 1.517115e-05<br />BrainRegion: sACC",".crossTalkKey: R14296_H7JLCBBXX<br />mitoRate: 0.01297116<br />rRNA_rate: 5.136608e-06<br />BrainRegion: sACC",".crossTalkKey: R14304_H7JKMBBXX<br />mitoRate: 0.01754904<br />rRNA_rate: 5.682725e-06<br />BrainRegion: sACC",".crossTalkKey: R14995_HFFGHBBXX<br />mitoRate: 0.01680911<br />rRNA_rate: 4.219755e-04<br />BrainRegion: sACC",".crossTalkKey: R15016_HF5JNBBXX<br />mitoRate: 0.01406832<br />rRNA_rate: 3.753388e-05<br />BrainRegion: sACC",".crossTalkKey: R15068_HFFGHBBXX<br />mitoRate: 0.01691871<br />rRNA_rate: 4.045060e-06<br />BrainRegion: sACC",".crossTalkKey: R15134_HFFGHBBXX<br />mitoRate: 0.01533612<br />rRNA_rate: 7.097073e-06<br />BrainRegion: sACC"],"key":["R14091_HFY2MBBXX","R14102_H7K5NBBXX","R14125_H7JM5BBXX","R14129_HCTYLBBXX","R14136_H7K5NBBXX","R14156_H7JHNBBXX","R14184_H7K5NBBXX","R14202_H7K5NBBXX","R14223_HF5JNBBXX","R14225_HF3JYBBXX","R14238_HF5JNBBXX","R14247_HF3JYBBXX","R14279_HFY2MBBXX","R14290_H7L3FBBXX","R14296_H7JLCBBXX","R14304_H7JKMBBXX","R14995_HFFGHBBXX","R15016_HF5JNBBXX","R15068_HFFGHBBXX","R15134_HFFGHBBXX"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(0,191,196,1)","opacity":1,"size":5.66929133858268,"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(0,191,196,1)"}},"hoveron":"points","set":"SharedData76ab4600","name":"sACC","legendgroup":"sACC","showlegend":true,"xaxis":"x2","yaxis":"y2","hoverinfo":"text","_isNestedKey":false,"frame":null}],"layout":{"xaxis":{"domain":[0,0.48],"automargin":true,"type":"linear","autorange":false,"range":[0.0110227062291971,0.0312894457739964],"tickmode":"array","ticktext":["0.015","0.020","0.025","0.030"],"tickvals":[0.015,0.02,0.025,0.03],"categoryorder":"array","categoryarray":["0.015","0.020","0.025","0.030"],"nticks":null,"ticks":"outside","tickcolor":"rgba(51,51,51,1)","ticklen":3.65296803652968,"tickwidth":0.66417600664176,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":11.689497716895},"tickangle":-0,"showline":false,"linecolor":null,"linewidth":0,"showgrid":true,"gridcolor":"rgba(255,255,255,1)","gridwidth":0.66417600664176,"zeroline":false,"anchor":"y","title":{"text":"mitoRate","font":{"color":"rgba(0,0,0,1)","family":"","size":14.6118721461187}},"hoverformat":".2f"},"xaxis2":{"domain":[0.52,1],"automargin":true,"type":"linear","autorange":false,"range":[0.0110227062291971,0.0312894457739964],"tickmode":"array","ticktext":["0.015","0.020","0.025","0.030"],"tickvals":[0.015,0.02,0.025,0.03],"categoryorder":"array","categoryarray":["0.015","0.020","0.025","0.030"],"nticks":null,"ticks":"outside","tickcolor":"rgba(51,51,51,1)","ticklen":3.65296803652968,"tickwidth":0.66417600664176,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":11.689497716895},"tickangle":-0,"showline":false,"linecolor":null,"linewidth":0,"showgrid":true,"gridcolor":"rgba(255,255,255,1)","gridwidth":0.66417600664176,"zeroline":false,"anchor":"y2","title":{"text":"mitoRate","font":{"color":"rgba(0,0,0,1)","family":"","size":14.6118721461187}},"hoverformat":".2f"},"yaxis2":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[-8.60895824462732e-05,0.00182337045043523],"tickmode":"array","ticktext":["0.0000","0.0005","0.0010","0.0015"],"tickvals":[0,0.0005,0.001,0.0015],"categoryorder":"array","categoryarray":["0.0000","0.0005","0.0010","0.0015"],"nticks":null,"ticks":"outside","tickcolor":"rgba(51,51,51,1)","ticklen":3.65296803652968,"tickwidth":0.66417600664176,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":11.689497716895},"tickangle":-0,"showline":false,"linecolor":null,"linewidth":0,"showgrid":true,"gridcolor":"rgba(255,255,255,1)","gridwidth":0.66417600664176,"zeroline":false,"anchor":"x2","hoverformat":".2f"},"yaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[0.0796948351929086,0.422887890350014],"tickmode":"array","ticktext":["0.1","0.2","0.3","0.4"],"tickvals":[0.1,0.2,0.3,0.4],"categoryorder":"array","categoryarray":["0.1","0.2","0.3","0.4"],"nticks":null,"ticks":"outside","tickcolor":"rgba(51,51,51,1)","ticklen":3.65296803652968,"tickwidth":0.66417600664176,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":11.689497716895},"tickangle":-0,"showline":false,"linecolor":null,"linewidth":0,"showgrid":true,"gridcolor":"rgba(255,255,255,1)","gridwidth":0.66417600664176,"zeroline":false,"anchor":"x","hoverformat":".2f"},"annotations":[{"text":"BrainRegion","x":0.4896,"y":1,"showarrow":false,"ax":0,"ay":0,"font":{"color":"rgba(0,0,0,1)","family":"","size":14.6118721461187},"xref":"paper","yref":"paper","textangle":-0,"xanchor":"left","yanchor":"bottom","legendTitle":true},{"text":"BrainRegion","x":1.0096,"y":1,"showarrow":false,"ax":0,"ay":0,"font":{"color":"rgba(0,0,0,1)","family":"","size":14.6118721461187},"xref":"paper","yref":"paper","textangle":-0,"xanchor":"left","yanchor":"bottom","legendTitle":true}],"shapes":[{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","x0":0,"x1":0.48,"y0":0,"y1":1},{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","x0":0.52,"x1":1,"y0":0,"y1":1}],"images":[],"margin":{"t":27.689497716895,"r":7.30593607305936,"b":41.6438356164384,"l":60.6392694063927},"plot_bgcolor":"rgba(235,235,235,1)","paper_bgcolor":"rgba(255,255,255,1)","font":{"color":"rgba(0,0,0,1)","family":"","size":14.6118721461187},"showlegend":true,"legend":{"bgcolor":"rgba(255,255,255,1)","bordercolor":"transparent","borderwidth":1.88976377952756,"font":{"color":"rgba(0,0,0,1)","family":"","size":11.689497716895},"y":0.96751968503937},"hovermode":"closest","barmode":"relative","dragmode":"zoom"},"attrs":{"27fe65c09d3a":{"key":{},"x":{},"y":{},"colour":{},"type":"scatter"},"27fe183a7b98":{"key":{},"x":{},"y":{},"colour":{},"type":"scatter"}},"source":"A","config":{"doubleClick":"reset","showSendToCloud":false},"highlight":{"on":"plotly_click","off":"plotly_doubleclick","persistent":true,"dynamic":true,"color":["rgba(228,26,28,1)","rgba(55,126,184,1)","rgba(77,175,74,1)","rgba(152,78,163,1)"],"selectize":true,"defaultValues":null,"opacityDim":0.2,"selected":{"opacity":1},"debounce":0,"ctGroups":["SharedData76ab4600"]},"subplot":true,"selectize":{"27fe11f55f54":{"items":[{"value":"R13896_H7JKMBBXX","label":"R13896_H7JKMBBXX"},{"value":"R13903_HCTYLBBXX","label":"R13903_HCTYLBBXX"},{"value":"R13904_H7K5NBBXX","label":"R13904_H7K5NBBXX"},{"value":"R13983_H7JKMBBXX","label":"R13983_H7JKMBBXX"},{"value":"R13985_H7JM5BBXX","label":"R13985_H7JM5BBXX"},{"value":"R13986_H7JM5BBXX","label":"R13986_H7JM5BBXX"},{"value":"R13995_H7JHNBBXX","label":"R13995_H7JHNBBXX"},{"value":"R14017_HF5JNBBXX","label":"R14017_HF5JNBBXX"},{"value":"R14023_H7JHNBBXX","label":"R14023_H7JHNBBXX"},{"value":"R14030_H7K5NBBXX","label":"R14030_H7K5NBBXX"},{"value":"R14031_HF5JNBBXX","label":"R14031_HF5JNBBXX"},{"value":"R14039_H7K5NBBXX","label":"R14039_H7K5NBBXX"},{"value":"R14069_HCTYLBBXX","label":"R14069_HCTYLBBXX"},{"value":"R14071_HF3JYBBXX","label":"R14071_HF3JYBBXX"},{"value":"R14073_HF3JYBBXX","label":"R14073_HF3JYBBXX"},{"value":"R14077_HCTYLBBXX","label":"R14077_HCTYLBBXX"},{"value":"R15058_HF5JNBBXX","label":"R15058_HF5JNBBXX"},{"value":"R15090_HF5JNBBXX","label":"R15090_HF5JNBBXX"},{"value":"R15093_HFY2MBBXX","label":"R15093_HFY2MBBXX"},{"value":"R15120_HFY2MBBXX","label":"R15120_HFY2MBBXX"},{"value":"R14091_HFY2MBBXX","label":"R14091_HFY2MBBXX"},{"value":"R14102_H7K5NBBXX","label":"R14102_H7K5NBBXX"},{"value":"R14125_H7JM5BBXX","label":"R14125_H7JM5BBXX"},{"value":"R14129_HCTYLBBXX","label":"R14129_HCTYLBBXX"},{"value":"R14136_H7K5NBBXX","label":"R14136_H7K5NBBXX"},{"value":"R14156_H7JHNBBXX","label":"R14156_H7JHNBBXX"},{"value":"R14184_H7K5NBBXX","label":"R14184_H7K5NBBXX"},{"value":"R14202_H7K5NBBXX","label":"R14202_H7K5NBBXX"},{"value":"R14223_HF5JNBBXX","label":"R14223_HF5JNBBXX"},{"value":"R14225_HF3JYBBXX","label":"R14225_HF3JYBBXX"},{"value":"R14238_HF5JNBBXX","label":"R14238_HF5JNBBXX"},{"value":"R14247_HF3JYBBXX","label":"R14247_HF3JYBBXX"},{"value":"R14279_HFY2MBBXX","label":"R14279_HFY2MBBXX"},{"value":"R14290_H7L3FBBXX","label":"R14290_H7L3FBBXX"},{"value":"R14296_H7JLCBBXX","label":"R14296_H7JLCBBXX"},{"value":"R14304_H7JKMBBXX","label":"R14304_H7JKMBBXX"},{"value":"R14995_HFFGHBBXX","label":"R14995_HFFGHBBXX"},{"value":"R15016_HF5JNBBXX","label":"R15016_HF5JNBBXX"},{"value":"R15068_HFFGHBBXX","label":"R15068_HFFGHBBXX"},{"value":"R15134_HFFGHBBXX","label":"R15134_HFFGHBBXX"}],"group":"SharedData76ab4600"}},"shinyEvents":["plotly_hover","plotly_click","plotly_selected","plotly_relayout","plotly_brushed","plotly_brushing","plotly_clickannotation","plotly_doubleclick","plotly_deselect","plotly_afterplot","plotly_sunburstclick"],"base_url":"https://plot.ly"},"evals":[],"jsHooks":[]}</script>
</div>
</div>
</div>
</div>
<div id="day-2" class="section level1" number="4">
<h1><span class="header-section-number">4</span> Day 2</h1>
<p>You can download the <a href="https://github.com/LieberInstitute/SPEAQeasy-example/blob/master/bootcamp_intro.Rmd"><strong>latest version of this Rmd document here</strong></a>.</p>
<div id="exploring-gene-expression" class="section level2" number="4.1">
<h2><span class="header-section-number">4.1</span> Exploring gene expression</h2>
<p>One way to explore gene expression data as a whole is to use a dimension reduction technique such as principal component analysis (PCA) or multidimensional scaling (MDS) <label for="sidenote-3" class="sidenote-number">3</label><label for="sidenote-3" class="margin-toggle" onclick="toggle_visibility('sidenote-3')">3</label><span class="sidenote" id="sidenote-3"><span class="sidenote-number">3</span> For details about them, check <a href="https://stats.stackexchange.com/questions/14002/whats-the-difference-between-principal-component-analysis-and-multidimensional">this thread on CrossValidated</a>.</span>. This is typically done with a subset of highly variable genes for computational speed, though with smaller datasets, you can do it with all the data.</p>
<p>In the <a href="http://research.libd.org/SPEAQeasy-example/de_analysis_speaqeasy.html#15_Explore_and_visualize_gene_expression">SPEAQeasy-example</a>, PCA was computed on the <code>log2(RPKM + 1)</code> data.</p>
<p>Once we have the top principal components (PCs), we can then make scatterplots or boxplots comparing them against some known covariates from our study as well as quality metrics generated by SPEAQeasy. It is also useful to compare the top PCs (the ones that explain the most variance) and color the samples (dots) by some grouping variables we are interested in; check <a href="http://bioconductor.org/packages/release/workflows/vignettes/rnaseqGene/inst/doc/rnaseqGene.html#pca-plot">this example</a> where the main covariate is associated with PC1 that explains ~50% of the variance. This process can reveal potential batch effects that we need to consider in our statistical models.</p>
<div id="exercise-1" class="section level3" number="4.1.1">
<h3><span class="header-section-number">4.1.1</span> Exercise</h3>
<ul>
<li>Compute the PCs just like in the <a href="http://research.libd.org/SPEAQeasy-example/de_analysis_speaqeasy.html#15_Explore_and_visualize_gene_expression">SPEAQeasy-example</a>.</li>
<li>Adapt the PC1 vs PC2 code to make three new scatterplots:
<ul>
<li>PC1 vs <code>mitoRate</code>,</li>
<li>PC1 vs <code>totalAssignedGene</code>,</li>
<li>PC1 vs <code>rRNA_rate</code></li>
</ul></li>
<li>Optional: make interactive versions of these graphs.</li>
</ul>
<p>Here’s some code to get you started.</p>
<pre class="r"><code>## Load some required packages
library(&quot;recount&quot;)
library(&quot;jaffelab&quot;)

# Filter for expressed
rse_gene &lt;- rse_gene[rowMeans(getRPKM(rse_gene, &quot;Length&quot;)) &gt; 0.2, ]

# Explore gene expression
geneExprs &lt;- log2(getRPKM(rse_gene, &quot;Length&quot;) + 1)
set.seed(20201006)
pca &lt;- prcomp(t(geneExprs))
pca_vars &lt;- getPcaVars(pca)
pca_vars_lab &lt;- paste0(
    &quot;PC&quot;, seq(along = pca_vars), &quot;: &quot;,
    pca_vars, &quot;% Var Expl&quot;
)</code></pre>
</div>
</div>
<div id="identifying-sample-swaps" class="section level2" number="4.2">
<h2><span class="header-section-number">4.2</span> Identifying sample swaps</h2>
<p>One of the main outputs from SPEAQeasy is a VCF file with ~740 common coding variants that you can compare across RNA-seq samples to identify issues with sample identity, or even better, against DNA genotyping data. <a href="http://research.libd.org/SPEAQeasy-example/swap_speaqeasy.html">SPEAQeasy-example</a> includes an example where we do the comparison and identify a few samples that are problematic.</p>
<p>Resolving sample identities is a headache-inducing process, since some issues are more complex than others to resolve and the decisions you make in how to prioritize the evidence will lead to different conclusions.</p>
</div>
<div id="statistical-modeling" class="section level2" number="4.3">
<h2><span class="header-section-number">4.3</span> Statistical modeling</h2>
<p>There are many Bioconductor packages that enable us to perform the statistical tests for differential expression analysis. Some of the main ones are <em><a href="https://bioconductor.org/packages/3.12/limma">limma</a></em>, <em><a href="https://bioconductor.org/packages/3.12/edgeR">edgeR</a></em>, and <em><a href="https://bioconductor.org/packages/3.12/DESeq2">DESeq2</a></em>. In our recent projects we have mostly used <em><a href="https://bioconductor.org/packages/3.12/limma">limma</a></em> due to its performance and speed, which matters most when analyzing exon or exon-exon junction expression data (count matrices with thousands of rows more than gene expression data).</p>
<div id="normalization" class="section level3" number="4.3.1">
<h3><span class="header-section-number">4.3.1</span> Normalization</h3>
<p>Prior to running a differential expression analysis, we first have to make the data comparable. This is done through a process called <em>normalization</em>. Many methods exist across Bioconductor packages for doing this. Given that we like to use <code>limma-voom</code> for the modeling, we’ll use the companion normalization methods from <em><a href="https://bioconductor.org/packages/3.12/edgeR">edgeR</a></em>. The <code>RNAseq123</code> Bioconductor workflow explains why normalization is important, so let’s <a href="http://bioconductor.org/packages/release/workflows/vignettes/RNAseq123/inst/doc/limmaWorkflow.html#normalising-gene-expression-distributions">take a look</a>.</p>
<p>Let’s use <code>edgeR::calcNormFactors()</code> with the SPEAQeasy-example data.</p>
<pre class="r"><code>## For the normalization functions
library(&quot;edgeR&quot;)

## From
## http://research.libd.org/SPEAQeasy-example/de_analysis_speaqeasy.html#16_Modeling
dge &lt;- DGEList(
    counts = assays(rse_gene)$counts,
    genes = rowData(rse_gene)
)
dge &lt;- calcNormFactors(dge)</code></pre>
</div>
<div id="design-matrix" class="section level3" number="4.3.2">
<h3><span class="header-section-number">4.3.2</span> Design matrix</h3>
<p>Differential expression analysis methods rely on the user specifying a <em>design matrix</em>. The simplest way to think about it is in terms of a linear regression, although some models use other distributions. R provides a useful function for making design matrices called <code>model.matrix()</code> that can take a <em>formula</em> input such as <code>~ group + lane</code> or <code>~ 0 + group + lane</code> as shown <a href="http://bioconductor.org/packages/release/workflows/vignettes/RNAseq123/inst/doc/limmaWorkflow.html#differential-expression-analysis">in RNAseq123</a>.</p>
<p>We can spend more time learning about regression starting with <a href="https://lcolladotor.github.io/bioc_team_ds/helping-others.html#linear-regression-example">linear regression</a>. However a user-friendly way that is quite powerful is implemented in the <em><a href="https://bioconductor.org/packages/3.12/ExploreModelMatrix">ExploreModelMatrix</a></em> package. Let’s look at the <a href="http://bioconductor.org/packages/release/bioc/vignettes/ExploreModelMatrix/inst/doc/ExploreModelMatrix.html">first example</a> from their vignette.</p>
<div id="exercise-2" class="section level4" number="4.3.2.1">
<h4><span class="header-section-number">4.3.2.1</span> Exercise</h4>
<ol style="list-style-type: decimal">
<li>Create the following model matrix.</li>
</ol>
<pre class="r"><code>## Clean up the PrimaryDx variable
rse_gene$PrimaryDx &lt;- factor(rse_gene$PrimaryDx)

## Modified from
## http://research.libd.org/SPEAQeasy-example/de_analysis_speaqeasy.html#16_Modeling
mod &lt;- model.matrix(~ PrimaryDx + BrainRegion,
    data = colData(rse_gene)
)</code></pre>
<ol start="2" style="list-style-type: decimal">
<li>Use <em><a href="https://bioconductor.org/packages/3.12/ExploreModelMatrix">ExploreModelMatrix</a></em> to visualize the design matrix.</li>
</ol>
<p>You might need to install <em><a href="https://bioconductor.org/packages/3.12/ExploreModelMatrix">ExploreModelMatrix</a></em> first as shown below.</p>
<pre class="r"><code>## Install a new package
BiocManager::install(&quot;ExploreModelMatrix&quot;)</code></pre>
<p>Then adapt the code from the <a href="http://bioconductor.org/packages/release/bioc/vignettes/ExploreModelMatrix/inst/doc/ExploreModelMatrix.html">first example</a> in their vignette documentation.</p>
<pre class="r"><code>library(&quot;ExploreModelMatrix&quot;)

## Adapt code from
## http://bioconductor.org/packages/release/bioc/vignettes/ExploreModelMatrix/inst/doc/ExploreModelMatrix.html</code></pre>
<p>Note that <code>sampleData</code> cannot take <code>NA</code> values, so you’ll have to choose some columns to work with. For example:</p>
<ul>
<li><code>PrimaryDx</code></li>
<li><code>BrainRegion</code></li>
<li><code>Sex</code></li>
<li><code>AgeDeath</code></li>
<li><code>mitoRate</code></li>
<li><code>totalAssignedGene</code></li>
<li><code>rRNA_rate</code></li>
</ul>
</div>
<div id="solution" class="section level4" number="4.3.2.2">
<h4><span class="header-section-number">4.3.2.2</span> Solution</h4>
<p>Here’s one solution to the exercise.</p>
<pre class="r"><code>## Here we select manually a few columns to work with
## and we use the same formula we used for making our model matrix
app &lt;-
    ExploreModelMatrix(
        sampleData = colData(rse_gene)[, c(
            &quot;PrimaryDx&quot;,
            &quot;BrainRegion&quot;,
            &quot;Sex&quot;,
            &quot;AgeDeath&quot;,
            &quot;mitoRate&quot;,
            &quot;totalAssignedGene&quot;,
            &quot;rRNA_rate&quot;
        )],
        designFormula = ~ PrimaryDx + BrainRegion
    )

## We can the explore the result interactively through a nice
## shiny web application created by ExploreModelMatrix
if (interactive()) {
      shiny::runApp(app)
  }</code></pre>
</div>
</div>
</div>
</div>
<div id="day-3" class="section level1" number="5">
<h1><span class="header-section-number">5</span> Day 3</h1>
<p>You can download the <a href="https://github.com/LieberInstitute/SPEAQeasy-example/blob/master/bootcamp_intro.Rmd"><strong>latest version of this Rmd document here</strong></a>.</p>
<div id="identifying-degs" class="section level2" number="5.1">
<h2><span class="header-section-number">5.1</span> Identifying DEGs</h2>
<p>Let’s take a look at the <code>RNAseq123</code> workflow to learn about how to <a href="http://bioconductor.org/packages/release/workflows/vignettes/RNAseq123/inst/doc/limmaWorkflow.html#removing-heteroscedascity-from-count-data">remove heteroscedascity from count data</a>.</p>
<pre class="r"><code>## Load limma
library(&quot;limma&quot;)

## Let&#39;s run limma-voom
vGene &lt;- invisible(voom(dge, mod, plot = TRUE))</code></pre>
<p><img src="bootcamp_intro_files/figure-html/run_voom-1.png" width="100%" /></p>
<p>Once we have removed the heteroscedascity from the data, we can proceed with the statistical tests. However, note that we have some repeated brains which means that we have non-independent data and need to take this into account.</p>
<pre class="r"><code>table(table(rse_gene$BrNum))</code></pre>
<pre><code>## 
##  1  2 
## 36  2</code></pre>
<p>We can do so using the function <code>limma::duplicateCorrelation()</code>.</p>
<pre class="r"><code># Get duplicate correlation
gene_dupCorr &lt;- duplicateCorrelation(vGene$E, mod,
    block = colData(rse_gene)$BrNum
)
gene_dupCorr$consensus.correlation</code></pre>
<pre><code>## [1] 0.6093648</code></pre>
<p>In this particular case, <code>duplicateCorrelation()</code> had several warnings, but in larger studies where we have more samples across individuals it will be more precise.</p>
<p>In any case, we can next finally perform the statistical tests using a series of <code>limma</code> functions: <code>lmFit()</code>, <code>eBayes()</code> and <code>topTable()</code>. Some key arguments are:</p>
<ul>
<li><code>lmFit(correlation, block)</code> which you won’t need if you don’t have a blocking covariate (duplicated brains or some other technical covariate like batch).</li>
<li><code>topTable(coef)</code> which specifies which coefficient we want to test. If we specify more than one, we will compute an F-statistic instead of a t-statistic. The other arguments we specify below allow us to get the full table of results instead of just the very first.</li>
</ul>
<pre class="r"><code># Fit linear model
fitGeneDupl &lt;- lmFit(
    vGene,
    correlation = gene_dupCorr$consensus.correlation,
    block = colData(rse_gene)$BrNum
)

# Here we perform an empirical Bayesian calculation to obtain
# our significant genes
ebGeneDupl &lt;- eBayes(fitGeneDupl)
outGeneDupl &lt;- topTable(
    ebGeneDupl,
    coef = 2,
    p.value = 1,
    number = nrow(rse_gene),
    sort.by = &quot;none&quot;
)
dim(outGeneDupl)</code></pre>
<pre><code>## [1] 26708    16</code></pre>
<pre class="r"><code>## Compare against the default topTable() output
## (we&#39;ll skip a few columns of output)
topTable(ebGeneDupl, coef = 2)[, -c(1, 3, 6:10)]</code></pre>
<pre><code>##                             gencodeID      gene_type   Symbol      logFC   AveExpr         t      P.Value  adj.P.Val
## ENSG00000126368.6   ENSG00000126368.6 protein_coding    NR1D1 -0.7598246 4.6128957 -5.135828 6.860966e-06 0.09162135
## ENSG00000269699.6   ENSG00000269699.6 protein_coding     ZIM2 -1.0460422 0.8970203 -5.363785 3.254603e-06 0.08692394
## ENSG00000249669.10 ENSG00000249669.10         lncRNA    CARMN  1.1888464 3.6042017  4.639830 3.404245e-05 0.26181040
## ENSG00000121858.11 ENSG00000121858.11 protein_coding  TNFSF10  1.3768752 2.2973424  4.595507 3.921078e-05 0.26181040
## ENSG00000135407.10 ENSG00000135407.10 protein_coding     AVIL  0.7222378 2.1327584  4.378458 7.793290e-05 0.35136699
## ENSG00000148835.11 ENSG00000148835.11 protein_coding     TAF5  0.3499746 2.3099665  4.339170 8.816211e-05 0.35136699
## ENSG00000124160.12 ENSG00000124160.12 protein_coding    NCOA5  0.3283855 4.4090971  4.160033 1.540310e-04 0.35982366
## ENSG00000125772.13 ENSG00000125772.13 protein_coding   GPCPD1  0.4943196 5.4346345  4.137133 1.653298e-04 0.35982366
## ENSG00000178878.12 ENSG00000178878.12 protein_coding   APOLD1 -1.3533527 5.2542372 -4.135982 1.659181e-04 0.35982366
## ENSG00000184988.8   ENSG00000184988.8 protein_coding TMEM106A  0.7631362 2.1042331  4.236908 1.213411e-04 0.35982366
##                            B
## ENSG00000126368.6  3.5828841
## ENSG00000269699.6  3.0822100
## ENSG00000249669.10 2.0883191
## ENSG00000121858.11 1.7155583
## ENSG00000135407.10 1.1570861
## ENSG00000148835.11 1.1047588
## ENSG00000124160.12 0.8633953
## ENSG00000125772.13 0.8167129
## ENSG00000178878.12 0.8099601
## ENSG00000184988.8  0.8026710</code></pre>
</div>
<div id="visualizing-degs" class="section level2" number="5.2">
<h2><span class="header-section-number">5.2</span> Visualizing DEGs</h2>
<p>Once we have our list of differentially expressed genes (DEGs), we can make some boxplots as in <a href="http://research.libd.org/SPEAQeasy-example/de_analysis_speaqeasy.html#17_Check_plots">SPEAQeasy-example</a> to check that the model results make sense. We frequently use <a href="http://research.libd.org/jaffelab/reference/cleaningY.html"><code>jaffelab::cleaningY()</code></a> to remove (regress out) the effect of covariates in our model.</p>
<p>Another way to visualize DEGs is to make a heatmap which is where <em><a href="https://CRAN.R-project.org/package=pheatmap">pheatmap</a></em> comes into play. It can be useful to see how the top genes are related.</p>
<pre class="r"><code>## For making a heatmap
library(&quot;pheatmap&quot;)

## Find the genes with a p-value less than 0.005
## (normally we would look at the FDR &lt; 0.05 genes or 0.1)
sigGene &lt;- outGeneDupl[outGeneDupl$P.Value &lt; 0.005, ]

## Extract the normalized expression
exprs_heatmap &lt;- vGene$E[rownames(sigGene), ]

## Build an annotation data.frame for our heatmap
df &lt;- as.data.frame(colData(rse_gene)[, c(&quot;PrimaryDx&quot;, &quot;Sex&quot;, &quot;BrainRegion&quot;)])
rownames(df) &lt;- colnames(exprs_heatmap) &lt;- gsub(&quot;_.*&quot;, &quot;&quot;, colnames(exprs_heatmap))
colnames(df) &lt;- c(&quot;Diagnosis&quot;, &quot;Sex&quot;, &quot;Region&quot;)

#  Manually determine coloring for plot annotation
palette_names = c(&#39;Dark2&#39;, &#39;Paired&#39;, &#39;YlOrRd&#39;)
ann_colors = list()
for (i in 1:ncol(df)) {
    col_name = colnames(df)[i]
    n_uniq_colors = length(unique(df[,col_name]))
    
    #   Use a unique palette with the correct number of levels, named with
    #   those levels
    ann_colors[[col_name]] = RColorBrewer::brewer.pal(n_uniq_colors, palette_names[i])[1:n_uniq_colors]
    names(ann_colors[[col_name]]) = unique(df[,col_name])
}</code></pre>
<pre><code>## Warning in RColorBrewer::brewer.pal(n_uniq_colors, palette_names[i]): minimal value for n is 3, returning requested palette with 3 different levels

## Warning in RColorBrewer::brewer.pal(n_uniq_colors, palette_names[i]): minimal value for n is 3, returning requested palette with 3 different levels

## Warning in RColorBrewer::brewer.pal(n_uniq_colors, palette_names[i]): minimal value for n is 3, returning requested palette with 3 different levels</code></pre>
<pre class="r"><code># Display heatmap
pheatmap(
    exprs_heatmap,
    cluster_rows = TRUE,
    show_rownames = FALSE,
    cluster_cols = TRUE,
    annotation_col = df,
    annotation_colors = ann_colors
)</code></pre>
<p><img src="bootcamp_intro_files/figure-html/pheatmap-1.png" width="100%" /></p>
<div id="exercise-3" class="section level3" number="5.2.1">
<h3><span class="header-section-number">5.2.1</span> Exercise</h3>
<ul>
<li>Make a plot with 4 boxplots (brain region by primary diagnosis) for the top differentially expressed gene.</li>
</ul>
</div>
<div id="solution-1" class="section level3" number="5.2.2">
<h3><span class="header-section-number">5.2.2</span> Solution</h3>
<pre class="r"><code>## Let&#39;s find that top gene
i &lt;- which.min(outGeneDupl$P.Value)

## We can use boxplot() from base R plots
boxplot(
    vGene$E[i,] ~ rse_gene$PrimaryDx + rse_gene$BrainRegion,
    las = 2,
    ylab = &quot;Expression&quot;,
    xlab = &quot;&quot;
)</code></pre>
<p><img src="bootcamp_intro_files/figure-html/top_gene_solution-1.png" width="100%" /></p>
<pre class="r"><code>## Or we can use ggplot2
## First we need to build a temporary data frame with
## the data for ggplot2
df_temp &lt;- data.frame(
    Expression = vGene$E[i,],
    Region = rse_gene$BrainRegion,
    Diagnosis = rse_gene$PrimaryDx
)
## Next we can make the boxplot, we&#39;ll use &quot;fill&quot; to color
## the boxes by the primary diagnosis variable
ggplot(df_temp, aes(y = Expression, x = Region, fill = Diagnosis)) +
    geom_boxplot() +
    theme_dark(base_size = 20)</code></pre>
<p><img src="bootcamp_intro_files/figure-html/top_gene_solution-2.png" width="100%" /></p>
</div>
</div>
<div id="go-enrichment-analyses" class="section level2" number="5.3">
<h2><span class="header-section-number">5.3</span> GO enrichment analyses</h2>
<p>Josh Stolz will show us some <a href="https://docs.google.com/presentation/d/1sQdOThDTq1iu8SV8H-HOJ8nGwgTxd5NSXnyfR1w6c28/edit?usp=sharing">slides</a>.</p>
<div id="demo" class="section level3" number="5.3.1">
<h3><span class="header-section-number">5.3.1</span> Demo</h3>
<p>You might need to install <em><a href="https://bioconductor.org/packages/3.12/enrichplot">enrichplot</a></em> first as shown below.</p>
<pre class="r"><code>## Install a new package
BiocManager::install(&quot;enrichplot&quot;)</code></pre>
<p>Here’s code for a GO demonstration by Josh.</p>
<pre class="r"><code>## Load required R packages
library(&quot;enrichplot&quot;)
library(&quot;clusterProfiler&quot;)
library(&quot;org.Hs.eg.db&quot;)

## Sample Data
data(geneList, package = &quot;DOSE&quot;)

## make cutoff log2 fold change
gene &lt;- names(geneList)[abs(geneList) &gt; 2]

## We need get different kinds of gene labels
gene.df &lt;- bitr(
    gene,
    fromType = &quot;ENTREZID&quot;,
    toType = c(&quot;ENSEMBL&quot;, &quot;SYMBOL&quot;),
    OrgDb = org.Hs.eg.db
)</code></pre>
<pre><code>## &#39;select()&#39; returned 1:many mapping between keys and columns</code></pre>
<pre><code>## Warning in bitr(gene, fromType = &quot;ENTREZID&quot;, toType = c(&quot;ENSEMBL&quot;, &quot;SYMBOL&quot;), : 0.48% of input gene IDs are fail to
## map...</code></pre>
<pre class="r"><code>## do gene ontology enrichment for the CC ontology
ego &lt;- enrichGO(
    gene = gene,
    universe = names(geneList),
    OrgDb = org.Hs.eg.db,
    ont = &quot;CC&quot;,
    pAdjustMethod = &quot;BH&quot;,
    pvalueCutoff = 0.01,
    qvalueCutoff = 0.05,
    readable = TRUE
)

## Visualize the GO enrichment results in different ways
barplot(ego, showCategory = 20)</code></pre>
<p><img src="bootcamp_intro_files/figure-html/go_demo-1.png" width="100%" /></p>
<pre class="r"><code>dotplot(ego, showCategory = 30)</code></pre>
<pre><code>## wrong orderBy parameter; set to default `orderBy = &quot;x&quot;`</code></pre>
<p><img src="bootcamp_intro_files/figure-html/go_demo-2.png" width="100%" /></p>
<pre class="r"><code>edox &lt;- setReadable(ego, &quot;org.Hs.eg.db&quot;, &quot;ENTREZID&quot;)

## cnetplot() and heatplot() show the actual genes
## that are present in our data (geneList) for each
## of the GO terms that were significantly enriched
cnetplot(edox,
    foldChange = geneList,
    circular = TRUE,
    colorEdge = TRUE)</code></pre>
<p><img src="bootcamp_intro_files/figure-html/go_demo-3.png" width="100%" /></p>
<pre class="r"><code>heatplot(edox, foldChange = geneList)</code></pre>
<p><img src="bootcamp_intro_files/figure-html/go_demo-4.png" width="100%" /></p>
</div>
<div id="exercise-4" class="section level3" number="5.3.2">
<h3><span class="header-section-number">5.3.2</span> Exercise</h3>
<ul>
<li>Which genes from our dataset are involved in the <em>chromosome, centromeric region</em> ontology?</li>
<li>Make a <code>heatplot()</code> using the results of enrichment analysis from the biological process ontology.</li>
</ul>
</div>
<div id="solution-2" class="section level3" number="5.3.3">
<h3><span class="header-section-number">5.3.3</span> Solution</h3>
<pre class="r"><code>## Finding the genes for _chromosome, centromeric region_
ego_df &lt;- as.data.frame(ego)
strsplit(
    ego_df$geneID[
        ego_df$Description == &quot;chromosome, centromeric region&quot;
        ],
    &quot;/&quot;)[[1]]</code></pre>
<pre><code>##  [1] &quot;CDCA8&quot;  &quot;CENPE&quot;  &quot;NDC80&quot;  &quot;HJURP&quot;  &quot;SKA1&quot;   &quot;NEK2&quot;   &quot;CENPM&quot;  &quot;CENPN&quot;  &quot;ERCC6L&quot; &quot;MAD2L1&quot; &quot;KIF18A&quot; &quot;CDT1&quot;  
## [13] &quot;BIRC5&quot;  &quot;TTK&quot;    &quot;NCAPG&quot;  &quot;AURKB&quot;  &quot;AURKA&quot;  &quot;CCNB1&quot;</code></pre>
<pre class="r"><code>## Run the enrichment with the BP ontology
ego_bp &lt;- enrichGO(
    gene = gene,
    universe = names(geneList),
    OrgDb = org.Hs.eg.db,
    ont = &quot;BP&quot;,
    pAdjustMethod = &quot;BH&quot;,
    pvalueCutoff = 0.01,
    qvalueCutoff = 0.05,
    readable = TRUE
)

## Then make the heatplot
ego_bp_x &lt;- setReadable(ego_bp, &quot;org.Hs.eg.db&quot;, &quot;ENTREZID&quot;)
heatplot(ego_bp_x, foldChange = geneList)</code></pre>
<p><img src="bootcamp_intro_files/figure-html/go_solution-1.png" width="100%" /></p>
</div>
</div>
<div id="beyond-speaqeasy" class="section level2" number="5.4">
<h2><span class="header-section-number">5.4</span> Beyond SPEAQeasy</h2>
<ul>
<li><a href="http://research.libd.org/recount3/">recount3</a>: over 750,000 human and mouse public RNA-seq processed with another aligner from the one used in SPEAQeasy: STAR instead of HISAT2.</li>
<li>Other tools like Salmon and kallisto.</li>
</ul>
</div>
</div>
<div id="reproducibility" class="section level1" number="6">
<h1><span class="header-section-number">6</span> Reproducibility</h1>
<pre class="r"><code># Date this report was generated
message(Sys.time())</code></pre>
<pre><code>## 2024-12-11 16:57:36</code></pre>
<pre class="r"><code># Reproducibility info
options(width = 120)
sessioninfo::session_info()</code></pre>
<pre><code>## ─ Session info ───────────────────────────────────────────────────────────────────────────────────────────────────────
##  setting  value                       
##  version  R version 4.0.3 (2020-10-10)
##  os       Ubuntu 20.04 LTS            
##  system   x86_64, linux-gnu           
##  ui       X11                         
##  language (EN)                        
##  collate  en_US.UTF-8                 
##  ctype    en_US.UTF-8                 
##  tz       Etc/UTC                     
##  date     2024-12-11                  
## 
## ─ Packages ───────────────────────────────────────────────────────────────────────────────────────────────────────────
##  package              * version  date       lib source                                   
##  AnnotationDbi        * 1.52.0   2020-10-27 [1] Bioconductor                             
##  askpass                1.1      2019-01-13 [2] RSPM (R 4.0.3)                           
##  assertthat             0.2.1    2019-03-21 [2] RSPM (R 4.0.3)                           
##  backports              1.2.1    2020-12-09 [1] RSPM (R 4.0.3)                           
##  base64enc              0.1-3    2015-07-28 [2] RSPM (R 4.0.3)                           
##  Biobase              * 2.50.0   2020-10-27 [1] Bioconductor                             
##  BiocFileCache          1.14.0   2020-10-27 [1] Bioconductor                             
##  BiocGenerics         * 0.36.1   2021-04-16 [1] Bioconductor                             
##  BiocManager            1.30.10  2019-11-16 [1] RSPM (R 4.0.0)                           
##  BiocParallel           1.24.1   2020-11-06 [1] Bioconductor                             
##  BiocStyle            * 2.18.1   2020-11-24 [1] Bioconductor                             
##  biomaRt                2.46.3   2021-02-09 [1] Bioconductor                             
##  Biostrings             2.58.0   2020-10-27 [1] Bioconductor                             
##  bit                    4.0.4    2020-08-04 [1] RSPM (R 4.0.5)                           
##  bit64                  4.0.5    2020-08-30 [1] RSPM (R 4.0.5)                           
##  bitops                 1.0-6    2013-08-17 [1] RSPM (R 4.0.3)                           
##  blob                   1.2.1    2020-01-20 [1] RSPM (R 4.0.3)                           
##  bookdown               0.21     2020-10-13 [1] RSPM (R 4.0.2)                           
##  BSgenome               1.58.0   2020-10-27 [1] Bioconductor                             
##  bslib                  0.2.4    2021-01-25 [1] RSPM (R 4.0.3)                           
##  bumphunter             1.32.0   2020-10-27 [1] Bioconductor                             
##  cachem                 1.0.4    2021-02-13 [2] RSPM (R 4.0.3)                           
##  checkmate              2.0.0    2020-02-06 [1] RSPM (R 4.0.3)                           
##  cli                    2.3.0    2021-01-31 [2] RSPM (R 4.0.3)                           
##  cluster                2.1.1    2021-02-14 [3] RSPM (R 4.0.3)                           
##  clusterProfiler      * 3.18.1   2021-02-09 [1] Bioconductor                             
##  codetools              0.2-18   2020-11-04 [3] RSPM (R 4.0.3)                           
##  colorspace             2.0-0    2020-11-11 [1] RSPM (R 4.0.3)                           
##  cowplot                1.1.1    2020-12-30 [1] RSPM (R 4.0.5)                           
##  crayon                 1.4.1    2021-02-08 [2] RSPM (R 4.0.3)                           
##  crosstalk              1.1.1    2021-01-12 [2] RSPM (R 4.0.3)                           
##  curl                   4.3      2019-12-02 [2] RSPM (R 4.0.3)                           
##  data.table             1.13.6   2020-12-30 [1] RSPM (R 4.0.3)                           
##  DBI                    1.1.1    2021-01-15 [1] RSPM (R 4.0.3)                           
##  dbplyr                 2.1.0    2021-02-03 [1] RSPM (R 4.0.3)                           
##  DelayedArray           0.16.3   2021-03-24 [1] Bioconductor                             
##  derfinder              1.24.2   2020-12-18 [1] Bioconductor                             
##  derfinderHelper        1.24.1   2020-12-18 [1] Bioconductor                             
##  digest                 0.6.27   2020-10-24 [2] RSPM (R 4.0.3)                           
##  DO.db                  2.9      2024-12-11 [1] Bioconductor                             
##  doRNG                  1.8.2    2020-01-27 [1] RSPM (R 4.0.3)                           
##  DOSE                   3.16.0   2020-10-27 [1] Bioconductor                             
##  downloader             0.4      2015-07-09 [1] RSPM (R 4.0.5)                           
##  dplyr                  1.0.4    2021-02-02 [1] RSPM (R 4.0.3)                           
##  DT                     0.17     2021-01-06 [2] RSPM (R 4.0.3)                           
##  edgeR                * 3.32.1   2021-01-14 [1] Bioconductor                             
##  ellipsis               0.3.1    2020-05-15 [2] RSPM (R 4.0.3)                           
##  enrichplot           * 1.10.2   2021-01-28 [1] Bioconductor                             
##  evaluate               0.14     2019-05-28 [2] RSPM (R 4.0.3)                           
##  ExploreModelMatrix   * 1.2.0    2020-10-27 [1] Bioconductor                             
##  farver                 2.0.3    2020-01-16 [1] RSPM (R 4.0.3)                           
##  fastmap                1.1.0    2021-01-25 [2] RSPM (R 4.0.3)                           
##  fastmatch              1.1-0    2017-01-28 [1] RSPM (R 4.0.3)                           
##  fgsea                  1.16.0   2020-10-27 [1] Bioconductor                             
##  foreach                1.5.1    2020-10-15 [1] RSPM (R 4.0.3)                           
##  foreign                0.8-81   2020-12-22 [3] RSPM (R 4.0.3)                           
##  Formula                1.2-4    2020-10-16 [1] RSPM (R 4.0.3)                           
##  generics               0.1.0    2020-10-31 [1] RSPM (R 4.0.3)                           
##  GenomeInfoDb         * 1.26.7   2021-04-08 [1] Bioconductor                             
##  GenomeInfoDbData       1.2.4    2024-12-11 [1] Bioconductor                             
##  GenomicAlignments      1.26.0   2020-10-27 [1] Bioconductor                             
##  GenomicFeatures        1.42.3   2021-04-01 [1] Bioconductor                             
##  GenomicFiles           1.26.0   2020-10-27 [1] Bioconductor                             
##  GenomicRanges        * 1.42.0   2020-10-27 [1] Bioconductor                             
##  GEOquery               2.58.0   2020-10-27 [1] Bioconductor                             
##  ggforce                0.3.2    2020-06-23 [1] RSPM (R 4.0.3)                           
##  ggnewscale             0.4.5    2021-01-11 [1] RSPM (R 4.0.3)                           
##  ggplot2              * 3.3.3    2020-12-30 [1] RSPM (R 4.0.3)                           
##  ggraph                 2.0.4    2020-11-16 [1] RSPM (R 4.0.3)                           
##  ggrepel                0.9.1    2021-01-15 [1] RSPM (R 4.0.3)                           
##  glue                   1.4.2    2020-08-27 [2] RSPM (R 4.0.3)                           
##  GO.db                  3.12.1   2024-12-11 [1] Bioconductor                             
##  googledrive            1.0.1    2020-05-05 [1] RSPM (R 4.0.0)                           
##  GOSemSim               2.16.1   2020-10-29 [1] Bioconductor                             
##  graphlayouts           0.7.1    2020-10-26 [1] RSPM (R 4.0.3)                           
##  gridExtra              2.3      2017-09-09 [1] RSPM (R 4.0.5)                           
##  gtable                 0.3.0    2019-03-25 [1] RSPM (R 4.0.3)                           
##  here                 * 1.0.1    2020-12-13 [1] RSPM (R 4.0.5)                           
##  highr                  0.8      2019-03-20 [2] RSPM (R 4.0.3)                           
##  Hmisc                  4.4-2    2020-11-29 [1] RSPM (R 4.0.3)                           
##  hms                    1.0.0    2021-01-13 [1] RSPM (R 4.0.3)                           
##  htmlTable              2.1.0    2020-09-16 [1] RSPM (R 4.0.3)                           
##  htmltools              0.5.1.1  2021-01-22 [2] RSPM (R 4.0.3)                           
##  htmlwidgets            1.5.3    2020-12-10 [2] RSPM (R 4.0.3)                           
##  httpuv                 1.5.5    2021-01-13 [1] RSPM (R 4.0.3)                           
##  httr                   1.4.2    2020-07-20 [2] RSPM (R 4.0.3)                           
##  igraph                 1.2.6    2020-10-06 [1] RSPM (R 4.0.3)                           
##  IRanges              * 2.24.1   2020-12-12 [1] Bioconductor                             
##  iterators              1.0.13   2020-10-15 [1] RSPM (R 4.0.3)                           
##  jaffelab             * 0.99.34  2024-12-11 [1] Github (LieberInstitute/jaffelab@a68c146)
##  jpeg                   0.1-8.1  2019-10-24 [1] RSPM (R 4.0.3)                           
##  jquerylib              0.1.3    2020-12-17 [1] RSPM (R 4.0.3)                           
##  jsonlite               1.7.2    2020-12-09 [2] RSPM (R 4.0.3)                           
##  knitr                  1.31     2021-01-27 [2] RSPM (R 4.0.3)                           
##  labeling               0.4.2    2020-10-20 [1] RSPM (R 4.0.3)                           
##  later                  1.1.0.1  2020-06-05 [2] RSPM (R 4.0.3)                           
##  lattice                0.20-41  2020-04-02 [3] CRAN (R 4.0.3)                           
##  latticeExtra           0.6-29   2019-12-19 [1] RSPM (R 4.0.3)                           
##  lazyeval               0.2.2    2019-03-15 [2] RSPM (R 4.0.3)                           
##  lifecycle              1.0.0    2021-02-15 [2] RSPM (R 4.0.3)                           
##  limma                * 3.46.0   2020-10-27 [1] Bioconductor                             
##  locfit                 1.5-9.4  2020-03-25 [1] RSPM (R 4.0.3)                           
##  magrittr               2.0.1    2020-11-17 [2] RSPM (R 4.0.3)                           
##  MASS                   7.3-53.1 2021-02-12 [3] RSPM (R 4.0.3)                           
##  Matrix                 1.3-2    2021-01-06 [3] RSPM (R 4.0.3)                           
##  MatrixGenerics       * 1.2.1    2021-01-30 [1] Bioconductor                             
##  matrixStats          * 0.58.0   2021-01-29 [1] RSPM (R 4.0.3)                           
##  memoise                2.0.0    2021-01-26 [2] RSPM (R 4.0.3)                           
##  mime                   0.10     2021-02-13 [2] RSPM (R 4.0.3)                           
##  munsell                0.5.0    2018-06-12 [1] RSPM (R 4.0.3)                           
##  nnet                   7.3-15   2021-01-24 [3] RSPM (R 4.0.3)                           
##  openssl                1.4.3    2020-09-18 [2] RSPM (R 4.0.3)                           
##  org.Hs.eg.db         * 3.12.0   2024-12-11 [1] Bioconductor                             
##  pheatmap             * 1.0.12   2019-01-04 [1] RSPM (R 4.0.5)                           
##  pillar                 1.4.7    2020-11-20 [2] RSPM (R 4.0.3)                           
##  pkgconfig              2.0.3    2019-09-22 [2] RSPM (R 4.0.3)                           
##  plotly               * 4.9.3    2021-01-10 [1] RSPM (R 4.0.3)                           
##  plyr                   1.8.6    2020-03-03 [1] RSPM (R 4.0.3)                           
##  png                    0.1-7    2013-12-03 [1] RSPM (R 4.0.3)                           
##  polyclip               1.10-0   2019-03-14 [1] RSPM (R 4.0.3)                           
##  prettyunits            1.1.1    2020-01-24 [2] RSPM (R 4.0.3)                           
##  progress               1.2.2    2019-05-16 [1] RSPM (R 4.0.3)                           
##  promises               1.2.0.1  2021-02-11 [2] RSPM (R 4.0.3)                           
##  purrr                  0.3.4    2020-04-17 [2] RSPM (R 4.0.3)                           
##  qvalue                 2.22.0   2020-10-27 [1] Bioconductor                             
##  R6                     2.5.0    2020-10-28 [2] RSPM (R 4.0.3)                           
##  rafalib              * 1.0.0    2015-08-09 [1] RSPM (R 4.0.0)                           
##  rappdirs               0.3.3    2021-01-31 [2] RSPM (R 4.0.3)                           
##  RColorBrewer           1.1-2    2014-12-07 [1] RSPM (R 4.0.3)                           
##  Rcpp                   1.0.6    2021-01-15 [2] RSPM (R 4.0.3)                           
##  RCurl                  1.98-1.2 2020-04-18 [1] RSPM (R 4.0.3)                           
##  readr                  1.4.0    2020-10-05 [1] RSPM (R 4.0.3)                           
##  recount              * 1.16.1   2020-12-18 [1] Bioconductor                             
##  rentrez                1.2.3    2020-11-10 [1] RSPM (R 4.0.5)                           
##  reshape2               1.4.4    2020-04-09 [1] RSPM (R 4.0.3)                           
##  rintrojs               0.2.2    2019-05-29 [1] RSPM (R 4.0.0)                           
##  rlang                  0.4.10   2020-12-30 [2] RSPM (R 4.0.3)                           
##  rmarkdown              2.6      2020-12-14 [1] RSPM (R 4.0.3)                           
##  rngtools               1.5      2020-01-23 [1] RSPM (R 4.0.3)                           
##  rpart                  4.1-15   2019-04-12 [3] CRAN (R 4.0.3)                           
##  rprojroot              2.0.2    2020-11-15 [2] RSPM (R 4.0.3)                           
##  Rsamtools              2.6.0    2020-10-27 [1] Bioconductor                             
##  RSQLite                2.2.3    2021-01-24 [1] RSPM (R 4.0.3)                           
##  rstudioapi             0.13     2020-11-12 [2] RSPM (R 4.0.3)                           
##  rtracklayer            1.50.0   2020-10-27 [1] Bioconductor                             
##  rvcheck                0.1.8    2020-03-01 [1] RSPM (R 4.0.0)                           
##  S4Vectors            * 0.28.1   2020-12-09 [1] Bioconductor                             
##  sass                   0.3.1    2021-01-24 [1] RSPM (R 4.0.3)                           
##  scales                 1.1.1    2020-05-11 [1] RSPM (R 4.0.3)                           
##  scatterpie             0.1.5    2020-09-09 [1] RSPM (R 4.0.2)                           
##  segmented              1.3-2    2021-02-09 [1] RSPM (R 4.0.3)                           
##  sessioninfo            1.1.1    2018-11-05 [1] RSPM (R 4.0.3)                           
##  shadowtext             0.0.7    2019-11-06 [1] RSPM (R 4.0.0)                           
##  shiny                  1.6.0    2021-01-25 [1] RSPM (R 4.0.3)                           
##  shinydashboard         0.7.1    2018-10-17 [1] RSPM (R 4.0.3)                           
##  shinyjs                2.0.0    2020-09-09 [1] RSPM (R 4.0.3)                           
##  statmod                1.4.35   2020-10-19 [1] RSPM (R 4.0.3)                           
##  stringi                1.5.3    2020-09-09 [2] RSPM (R 4.0.3)                           
##  stringr                1.4.0    2019-02-10 [2] RSPM (R 4.0.3)                           
##  SummarizedExperiment * 1.20.0   2020-10-27 [1] Bioconductor                             
##  survival               3.2-7    2020-09-28 [3] CRAN (R 4.0.3)                           
##  tibble                 3.0.6    2021-01-29 [2] RSPM (R 4.0.3)                           
##  tidygraph              1.2.0    2020-05-12 [1] RSPM (R 4.0.3)                           
##  tidyr                  1.1.2    2020-08-27 [1] RSPM (R 4.0.3)                           
##  tidyselect             1.1.0    2020-05-11 [1] RSPM (R 4.0.3)                           
##  tweenr                 1.0.1    2018-12-14 [1] RSPM (R 4.0.3)                           
##  VariantAnnotation      1.36.0   2020-10-27 [1] Bioconductor                             
##  vctrs                  0.3.6    2020-12-17 [2] RSPM (R 4.0.3)                           
##  viridis                0.5.1    2018-03-29 [1] RSPM (R 4.0.3)                           
##  viridisLite            0.3.0    2018-02-01 [1] RSPM (R 4.0.3)                           
##  withr                  2.4.1    2021-01-26 [2] RSPM (R 4.0.3)                           
##  xfun                   0.21     2021-02-10 [2] RSPM (R 4.0.3)                           
##  XML                    3.99-0.5 2020-07-23 [1] RSPM (R 4.0.3)                           
##  xml2                   1.3.2    2020-04-23 [2] RSPM (R 4.0.3)                           
##  xtable                 1.8-4    2019-04-21 [1] RSPM (R 4.0.5)                           
##  XVector                0.30.0   2020-10-27 [1] Bioconductor                             
##  yaml                   2.2.1    2020-02-01 [2] RSPM (R 4.0.3)                           
##  zlibbioc               1.36.0   2020-10-27 [1] Bioconductor                             
## 
## [1] /__w/_temp/Library
## [2] /usr/local/lib/R/site-library
## [3] /usr/local/lib/R/library</code></pre>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("show" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": {
      styles: {
        ".MathJax_Display": {
           "text-align": "center",
           padding: "0px 150px 0px 65px",
           margin: "0px 0px 0.5em"
        },
        "@media screen and (max-width: 991px)": {
            ".MathJax_Display": {
               "text-align": "center",
               padding: "0 0 0 0"
            }
         }
      }
    }
  });
</script>
<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
