<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Differential Expression Analysis</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
      </style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">

</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>

<link rel="stylesheet" href="/usr/local/lib/R/host-site-library/BiocStyle/resources/html/bioconductor.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 828px;
  margin-left: auto;
  margin-right: auto;
}

img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {

}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 246px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



<script>
function toggle_visibility(id1) {
  var e = document.getElementById(id1);
  e.style.display = ((e.style.display!="none") ? "none" : "block");
}
</script>

</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">SPEAQeasy example</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="prepare_data.html">Download Example Data</a>
</li>
<li>
  <a href="run_speaqeasy.html">Run SPEAQeasy</a>
</li>
<li>
  <a href="swap_speaqeasy.html">Resolve Sample Identities</a>
</li>
<li>
  <a href="de_analysis_speaqeasy.html">Differential Expression</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/LieberInstitute/SPEAQeasy-example">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Differential Expression Analysis</h1>
<p class="author-name">Joshua M. Stolz<span class="affil-mark">1</span></p>
<p class="author-affiliation"><span class="affil-mark">1</span>Lieber Institute for Brain Development, Johns Hopkins Medical Campus</p>
<h4 class="date">9/18/2020</h4>

</div>


<div id="analysis" class="section level1">
<h1><span class="header-section-number">1</span> Analysis</h1>
<p>The following analysis explores a <code>RangedSummarizedExperiment</code> object from the SPEAQeasy pipeline. Note that we will use a modified version of the object, which resolved sample identity issues which were present in the raw output from SPEAQeasy. This object also includes phenotype data added after resolving identity issues. Though SPEAQeasy produces objects for several feature types (genes, exons, exon-exon junctions), we will demonstrate an example analysis for just genes. We will perform differential expression across some typical variables of interest (e.g. sex, age, race) and show how to perform principal component analysis (PCA) and visualize findings with plots.</p>
<div id="load-required-libraries" class="section level2">
<h2><span class="header-section-number">1.1</span> Load required libraries</h2>
<pre class="r"><code>library(&quot;SummarizedExperiment&quot;)
library(&quot;recount&quot;)
library(&quot;edgeR&quot;)
library(&quot;limma&quot;)
library(&quot;jaffelab&quot;) # GitHub: LieberInstitute/jaffelab
library(&quot;RColorBrewer&quot;)
library(&quot;clusterProfiler&quot;)
library(&quot;org.Hs.eg.db&quot;)
library(&quot;pheatmap&quot;)
library(&quot;here&quot;)
library(&quot;devtools&quot;)
library(&quot;BiocStyle&quot;)</code></pre>
</div>
<div id="load-data-and-prepare-directories-to-place-outputs-in" class="section level2">
<h2><span class="header-section-number">1.2</span> Load data and prepare directories to place outputs in</h2>
<p>For those who ran SPEAQeasy from the example FASTQ data set, the <code>RangedSummarizedExperiment</code> will have a different path, as specified with the <code>--output</code> flag.</p>
<pre class="r"><code># Load the RSE gene object
load(here(&quot;rse_speaqeasy.RData&quot;), verbose = TRUE)</code></pre>
<pre><code>## Loading objects:
##   rse_gene</code></pre>
<pre class="r"><code># Create directories to organize outputs from this analysis
dir.create(here(&quot;DE_analysis&quot;, &quot;pdfs&quot;), showWarnings = FALSE)
dir.create(here(&quot;DE_analysis&quot;, &quot;tables&quot;), showWarnings = FALSE)
dir.create(here(&quot;DE_analysis&quot;, &quot;rdas&quot;), showWarnings = FALSE)

## Clean up the PrimaryDx variable
rse_gene$PrimaryDx &lt;- factor(rse_gene$PrimaryDx)</code></pre>
</div>
<div id="statistics-pcs" class="section level2">
<h2><span class="header-section-number">1.3</span> statistics PCs</h2>
<p>Here we are using principal component analysis to control for the listed variables impact on expression. This will be later added into our linear model.</p>
<pre class="r"><code>col_names = c(&#39;trimmed&#39;, &#39;numReads&#39;, &#39;numMapped&#39;, &#39;numUnmapped&#39;, 
              &#39;overallMapRate&#39;, &#39;concordMapRate&#39;, &#39;totalMapped&#39;, &#39;mitoMapped&#39;,
              &#39;mitoRate&#39;, &#39;totalAssignedGene&#39;)
statsPca = prcomp(as.data.frame(colData(rse_gene)[,col_names]))
rse_gene$PC = statsPca$x[,1]
getPcaVars(statsPca)[1]</code></pre>
<pre><code>## [1] 87.3</code></pre>
</div>
<div id="stats-vs.-race-and-brain-region" class="section level2">
<h2><span class="header-section-number">1.4</span> Stats vs. race and brain region</h2>
<pre class="r"><code># Filter for expressed
rse_gene = rse_gene[rowMeans(getRPKM(rse_gene,&quot;Length&quot;)) &gt; 0.2,]

# Check if certain statistics changed by race or region

# Display box plots here
boxplot(rse_gene$rRNA_rate ~ rse_gene$BrainRegion,xlab=&quot;&quot;)</code></pre>
<p><img src="de_analysis_speaqeasy_files/figure-html/race_region-1.png" width="768" /></p>
<pre class="r"><code>boxplot(rse_gene$mitoRate ~ rse_gene$BrainRegion,xlab=&quot;&quot;)</code></pre>
<p><img src="de_analysis_speaqeasy_files/figure-html/race_region-2.png" width="768" /></p>
<pre class="r"><code>boxplot(rse_gene$gene_Assigned ~ rse_gene$BrainRegion,xlab=&quot;&quot;)</code></pre>
<p><img src="de_analysis_speaqeasy_files/figure-html/race_region-3.png" width="768" /></p>
<pre class="r"><code>boxplot(rse_gene$mitoRate ~ rse_gene$Race,las=3,xlab=&quot;&quot;)</code></pre>
<p><img src="de_analysis_speaqeasy_files/figure-html/race_region-4.png" width="768" /></p>
<pre class="r"><code>boxplot(rse_gene$gene_Assigned ~ rse_gene$Race,las=3,xlab=&quot;&quot;)</code></pre>
<p><img src="de_analysis_speaqeasy_files/figure-html/race_region-5.png" width="768" /></p>
<pre class="r"><code># Save box plots to PDF
pdf(file = here(&quot;DE_analysis&quot;, &quot;pdfs&quot;, &quot;Region_Race_cellcheck.pdf&quot;))
boxplot(rse_gene$rRNA_rate ~ rse_gene$BrainRegion,xlab=&quot;&quot;)
boxplot(rse_gene$mitoRate ~ rse_gene$BrainRegion,xlab=&quot;&quot;)
boxplot(rse_gene$gene_Assigned ~ rse_gene$BrainRegion,xlab=&quot;&quot;)
boxplot(rse_gene$mitoRate ~ rse_gene$Race,las=3,xlab=&quot;&quot;)
boxplot(rse_gene$gene_Assigned ~ rse_gene$Race,las=3,xlab=&quot;&quot;)
dev.off()</code></pre>
<pre><code>## png 
##   2</code></pre>
</div>
<div id="explore-and-visualize-gene-expression" class="section level2">
<h2><span class="header-section-number">1.5</span> Explore and visualize gene expression</h2>
<pre class="r"><code># Explore gene expression
geneExprs = log2(getRPKM(rse_gene,&quot;Length&quot;)+1)
pca = prcomp(t(geneExprs))
pca_vars = getPcaVars(pca)
pca_vars_lab = paste0(&quot;PC&quot;, seq(along=pca_vars), &quot;: &quot;,
    pca_vars, &quot;% Var Expl&quot;)

# Group together code for generating plots of interest
generate_plots = function() {
    par(mar=c(8,6,2,2),cex.axis=1.8,cex.lab=1.8)
    palette(brewer.pal(4,&quot;Dark2&quot;))
    
    # PC1 vs. PC2
    plot(pca$x, pch=21, bg=factor(rse_gene$PrimaryDx),cex=1.2,
        xlab = pca_vars_lab[1], ylab = pca_vars_lab[2])
    legend(&quot;bottomleft&quot;, levels(rse_gene$PrimaryDx), col=1:2, pch=15,cex=2)
    
    
    # By line
    for(i in 1:10) {
        boxplot(pca$x[,i] ~ rse_gene$Sex,
            ylab=pca_vars_lab[i], las = 3,xlab=&quot;Sex&quot;,outline=FALSE)
        points( pca$x[,i] ~ jitter(as.numeric(factor(rse_gene$Sex))),
            pch = 21, bg = rse_gene$PrimaryDx,cex=1.2)
    }
    
    # By experiment
    for(i in 1:10) {
        boxplot(pca$x[,i] ~ rse_gene$Race,
            ylab=pca_vars_lab[i], las = 3,xlab=&quot;Race&quot;,outline=FALSE)
        points( pca$x[,i] ~ jitter(as.numeric(factor(rse_gene$Race))),
            pch = 21, bg = rse_gene$PrimaryDx,cex=1.2)
    }
}

# Display plots
generate_plots()</code></pre>
<p><img src="de_analysis_speaqeasy_files/figure-html/gene_expression-1.png" width="768" /><img src="de_analysis_speaqeasy_files/figure-html/gene_expression-2.png" width="768" /><img src="de_analysis_speaqeasy_files/figure-html/gene_expression-3.png" width="768" /><img src="de_analysis_speaqeasy_files/figure-html/gene_expression-4.png" width="768" /><img src="de_analysis_speaqeasy_files/figure-html/gene_expression-5.png" width="768" /><img src="de_analysis_speaqeasy_files/figure-html/gene_expression-6.png" width="768" /><img src="de_analysis_speaqeasy_files/figure-html/gene_expression-7.png" width="768" /><img src="de_analysis_speaqeasy_files/figure-html/gene_expression-8.png" width="768" /><img src="de_analysis_speaqeasy_files/figure-html/gene_expression-9.png" width="768" /><img src="de_analysis_speaqeasy_files/figure-html/gene_expression-10.png" width="768" /><img src="de_analysis_speaqeasy_files/figure-html/gene_expression-11.png" width="768" /><img src="de_analysis_speaqeasy_files/figure-html/gene_expression-12.png" width="768" /><img src="de_analysis_speaqeasy_files/figure-html/gene_expression-13.png" width="768" /><img src="de_analysis_speaqeasy_files/figure-html/gene_expression-14.png" width="768" /><img src="de_analysis_speaqeasy_files/figure-html/gene_expression-15.png" width="768" /><img src="de_analysis_speaqeasy_files/figure-html/gene_expression-16.png" width="768" /><img src="de_analysis_speaqeasy_files/figure-html/gene_expression-17.png" width="768" /><img src="de_analysis_speaqeasy_files/figure-html/gene_expression-18.png" width="768" /><img src="de_analysis_speaqeasy_files/figure-html/gene_expression-19.png" width="768" /><img src="de_analysis_speaqeasy_files/figure-html/gene_expression-20.png" width="768" /><img src="de_analysis_speaqeasy_files/figure-html/gene_expression-21.png" width="768" /></p>
<pre class="r"><code># Write plots to PDF
pdf(here(&quot;DE_analysis&quot;, &quot;pdfs&quot;, &quot;PCA_plotsExprs.pdf&quot;), w=9)
generate_plots()
dev.off()</code></pre>
<pre><code>## png 
##   2</code></pre>
</div>
<div id="modeling" class="section level2">
<h2><span class="header-section-number">1.6</span> Modeling</h2>
<pre class="r"><code>dge = DGEList(counts = assays(rse_gene)$counts,
    genes = rowData(rse_gene))
dge = calcNormFactors(dge)

# Mean-variance
mod = model.matrix(~PrimaryDx + PC + BrainRegion,
    data=colData(rse_gene))

vGene = invisible(voom(dge,mod,plot=TRUE))</code></pre>
<p><img src="de_analysis_speaqeasy_files/figure-html/modeling-1.png" width="768" /></p>
<pre class="r"><code># Also write mean-variance plot to PDF
pdf(file = &quot;DE_analysis/pdfs/vGene.pdf&quot;)
invisible(voom(dge,mod,plot=TRUE))
dev.off()</code></pre>
<pre><code>## png 
##   2</code></pre>
<pre class="r"><code># Get duplicate correlation
gene_dupCorr = duplicateCorrelation(vGene$E, mod,
    block=colData(rse_gene)$SAMPLE_ID)
save(gene_dupCorr, file = here(&quot;DE_analysis&quot;, &quot;rdas&quot;, &quot;gene_dupCorr_neurons.rda&quot;))

# Fit linear model
fitGeneDupl = lmFit(vGene,
    correlation=gene_dupCorr$consensus.correlation,
    block=colData(rse_gene)$SAMPLEID)

# Here we perform an empirical Bayesian calculation to obtain our significant genes
ebGeneDupl = eBayes(fitGeneDupl)
outGeneDupl = topTable(ebGeneDupl,coef=2,
    p.value = 1,number=nrow(rse_gene),sort=&quot;none&quot;)

pdf(file = here(&quot;DE_analysis&quot;, &quot;pdfs&quot;, &quot;hist_pval.pdf&quot;))
hist(outGeneDupl$P.Value)
dev.off()</code></pre>
<pre><code>## png 
##   2</code></pre>
<pre class="r"><code>hist(outGeneDupl$P.Value)</code></pre>
<p><img src="de_analysis_speaqeasy_files/figure-html/modeling-2.png" width="768" /></p>
<pre class="r"><code>table(outGeneDupl$adj.P.Val &lt; 0.05)</code></pre>
<pre><code>## 
## FALSE  TRUE 
## 26707     1</code></pre>
<pre class="r"><code>table(outGeneDupl$adj.P.Val &lt; 0.1)</code></pre>
<pre><code>## 
## FALSE  TRUE 
## 26706     2</code></pre>
<pre class="r"><code>sigGeneDupl =  topTable(ebGeneDupl,coef=2,
    p.value = 0.1,number=nrow(rse_gene))

sigGeneDupl[,c(&quot;Symbol&quot;,&quot;logFC&quot;, &quot;P.Value&quot;,&quot;AveExpr&quot;)]</code></pre>
<pre><code>##                   Symbol      logFC      P.Value   AveExpr
## ENSG00000269699.6   ZIM2 -1.1238665 1.448916e-06 0.8970203
## ENSG00000126368.6  NR1D1 -0.7637626 6.833687e-06 4.6128957</code></pre>
<pre class="r"><code>sigGeneDupl[sigGeneDupl$logFC &gt; 0,c(&quot;Symbol&quot;,&quot;logFC&quot;, &quot;P.Value&quot;)]</code></pre>
<pre><code>## [1] Symbol  logFC   P.Value
## &lt;0 rows&gt; (or 0-length row.names)</code></pre>
<pre class="r"><code>sigGeneDupl[sigGeneDupl$logFC &lt;  0,c(&quot;Symbol&quot;,&quot;logFC&quot;, &quot;P.Value&quot;)]</code></pre>
<pre><code>##                   Symbol      logFC      P.Value
## ENSG00000269699.6   ZIM2 -1.1238665 1.448916e-06
## ENSG00000126368.6  NR1D1 -0.7637626 6.833687e-06</code></pre>
<pre class="r"><code>write.csv(outGeneDupl, file = here(&quot;DE_analysis&quot;, &quot;tables&quot;, &quot;de_stats_allExprs.csv&quot;))
write.csv(sigGeneDupl, file = here(&quot;DE_analysis&quot;, &quot;tables&quot;, &quot;de_stats_fdr10_sorted.csv&quot;))</code></pre>
</div>
<div id="check-plots" class="section level2">
<h2><span class="header-section-number">1.7</span> Check plots</h2>
<pre class="r"><code>exprs = vGene$E[rownames(sigGeneDupl),]

# Group together code for displaying boxplots
generate_plots = function() {
    par(mar=c(8,6,4,2),cex.axis=1.8,cex.lab=1.8, cex.main=1.8)
    palette(brewer.pal(4,&quot;Dark2&quot;))
    
    for(i in 1:nrow(sigGeneDupl)) {
        yy = exprs[i,]
        boxplot(yy ~ rse_gene$PrimaryDx, outline=FALSE,
            ylim=range(yy), ylab=&quot;Normalized log2 Exprs&quot;, xlab=&quot;&quot;,
        main = paste(sigGeneDupl$Symbol[i], &quot;-&quot;, sigGeneDupl$gencodeID[i]))
        points(yy ~ jitter(as.numeric(rse_gene$PrimaryDx)),
            pch = 21, bg= rse_gene$PrimaryDx,cex=1.3)
        ll = ifelse(sigGeneDupl$logFC[i] &gt; 0, &quot;topleft&quot;, &quot;topright&quot;)
        legend(ll, paste0(&quot;p=&quot;, signif(sigGeneDupl$P.Value[i],3)), cex=1.3)
    }
}

# Show boxplots
generate_plots()</code></pre>
<p><img src="de_analysis_speaqeasy_files/figure-html/check_plots-1.png" width="768" /><img src="de_analysis_speaqeasy_files/figure-html/check_plots-2.png" width="768" /></p>
<pre class="r"><code># Write plots to PDF
pdf(here(&quot;DE_analysis&quot;, &quot;pdfs&quot;, &quot;DE_boxplots_byDiagnosis.pdf&quot;), w=10)
generate_plots()
dev.off()</code></pre>
<pre><code>## png 
##   2</code></pre>
<pre class="r"><code>e = geneExprs[rownames(sigGeneDupl),]

generate_plots = function() {
    par(mar=c(8,6,4,2),cex.axis=1.8,cex.lab=1.8, cex.main=1.8)
    palette(brewer.pal(4,&quot;Dark2&quot;))
    for(i in 1:nrow(sigGeneDupl)) {
        yy = e[i,]
        boxplot(yy ~ rse_gene$PrimaryDx, las=3,outline=FALSE,
              ylim=range(yy), ylab=&quot;log2(RPKM+1)&quot;, xlab=&quot;&quot;,
        main = paste(sigGeneDupl$Symbol[i], &quot;-&quot;, sigGeneDupl$gencodeID[i]))
        points(yy ~ jitter(as.numeric(rse_gene$PrimaryDx)),
            pch = 21, bg= rse_gene$PrimaryDx,cex=1.3)
        ll = ifelse(sigGeneDupl$logFC[i] &gt; 0, &quot;topleft&quot;, &quot;topright&quot;)
        legend(ll, paste0(&quot;p=&quot;, signif(sigGeneDupl$P.Value[i],3)), cex=1.3)
    }
}

# Display plots
generate_plots()</code></pre>
<p><img src="de_analysis_speaqeasy_files/figure-html/check_plots-3.png" width="768" /><img src="de_analysis_speaqeasy_files/figure-html/check_plots-4.png" width="768" /></p>
<pre class="r"><code># Write the same plots to PDF
pdf(here(&quot;DE_analysis&quot;, &quot;pdfs&quot;, &quot;DE_boxplots_byGenome_log2RPKM.pdf&quot;), w=10)
generate_plots()
dev.off()</code></pre>
<pre><code>## png 
##   2</code></pre>
</div>
<div id="gene-ontology" class="section level2">
<h2><span class="header-section-number">1.8</span> Gene ontology</h2>
<p><code>clusterProfiler</code> is a gene ontology package we will use to see if our genes are specifically differentially expressed in certain pathways.</p>
<pre class="r"><code># Get significant genes by sign
sigGene = outGeneDupl[outGeneDupl$P.Value &lt; 0.005,]
sigGeneList = split(as.character(sigGene$EntrezID), sign(sigGene$logFC))
sigGeneList = lapply(sigGeneList, function(x) x[!is.na(x)])
geneUniverse = as.character(outGeneDupl$EntrezID)
geneUniverse = geneUniverse[!is.na(geneUniverse)]

# Do GO and KEGG
goBP_Adj &lt;- compareCluster(sigGeneList, fun = &quot;enrichGO&quot;,
    universe = geneUniverse, OrgDb = org.Hs.eg.db,
    ont = &quot;BP&quot;, pAdjustMethod = &quot;BH&quot;, pvalueCutoff  = 1,
    qvalueCutoff  = 1,  readable= TRUE)

goMF_Adj &lt;- compareCluster(sigGeneList, fun = &quot;enrichGO&quot;,
    universe = geneUniverse, OrgDb = org.Hs.eg.db,
    ont = &quot;MF&quot;, pAdjustMethod = &quot;BH&quot;, pvalueCutoff  = 1,
    qvalueCutoff  = 1,  readable= TRUE)

goCC_Adj &lt;- compareCluster(sigGeneList, fun = &quot;enrichGO&quot;,
    universe = geneUniverse, OrgDb = org.Hs.eg.db,
    ont = &quot;CC&quot;, pAdjustMethod = &quot;BH&quot;, pvalueCutoff  = 1,
    qvalueCutoff  = 1,  readable= TRUE)

kegg_Adj &lt;- compareCluster(sigGeneList, fun = &quot;enrichKEGG&quot;,
    universe = geneUniverse,  pAdjustMethod = &quot;BH&quot;,
    pvalueCutoff  = 1, qvalueCutoff  = 1)

save(goBP_Adj, goCC_Adj, goMF_Adj, kegg_Adj,
    file = here(&quot;DE_analysis&quot;, &quot;rdas&quot;, &quot;gene_set_objects_p005.rda&quot;))

goList = list(BP = goBP_Adj, MF = goMF_Adj, CC = goCC_Adj, KEGG = kegg_Adj)
goDf = dplyr::bind_rows(lapply(goList, as.data.frame), .id = &quot;Ontology&quot;)
goDf = goDf[order(goDf$pvalue),]

write.csv(goDf, file = here(&quot;DE_analysis&quot;, &quot;tables&quot;, &quot;geneSet_output.csv&quot;), row.names=FALSE)

options(width=130)
goDf[goDf$p.adjust &lt; 0.05, c(1:5,7)]</code></pre>
<pre><code>##      Ontology Cluster         ID                                                                  Description GeneRatio
## 1311       BP       1 GO:0032640                                             tumor necrosis factor production      9/84
## 1312       BP       1 GO:0071706                        tumor necrosis factor superfamily cytokine production      9/84
## 1313       BP       1 GO:0050865                                                regulation of cell activation     14/84
## 1314       BP       1 GO:0032611                                                interleukin-1 beta production      7/84
## 1315       BP       1 GO:0032612                                                     interleukin-1 production      7/84
## 1316       BP       1 GO:0032680                               regulation of tumor necrosis factor production      8/84
## 1317       BP       1 GO:1903555          regulation of tumor necrosis factor superfamily cytokine production      8/84
## 1318       BP       1 GO:0032651                                  regulation of interleukin-1 beta production      6/84
## 1319       BP       1 GO:0050867                                       positive regulation of cell activation     10/84
## 1320       BP       1 GO:0002521                                                    leukocyte differentiation     12/84
## 1321       BP       1 GO:0002694                                           regulation of leukocyte activation     12/84
## 1322       BP       1 GO:0032652                                       regulation of interleukin-1 production      6/84
## 1323       BP       1 GO:0009617                                                        response to bacterium     12/84
## 1324       BP       1 GO:0042742                                                defense response to bacterium      7/84
## 1325       BP       1 GO:0071346                                        cellular response to interferon-gamma      7/84
## 1326       BP       1 GO:0050900                                                          leukocyte migration     10/84
## 3316       MF       1 GO:0045028                    G protein-coupled purinergic nucleotide receptor activity      3/88
## 1327       BP       1 GO:0002687                                   positive regulation of leukocyte migration      6/84
## 1328       BP       1 GO:0034341                                                 response to interferon-gamma      7/84
## 3317       MF       1 GO:0046935                           1-phosphatidylinositol-3-kinase regulator activity      3/88
## 1329       BP       1 GO:0031347                                               regulation of defense response     12/84
## 1330       BP       1 GO:0002685                                            regulation of leukocyte migration      7/84
## 1331       BP       1 GO:0071219                            cellular response to molecule of bacterial origin      7/84
## 1332       BP       1 GO:0032760                      positive regulation of tumor necrosis factor production      5/84
## 1333       BP       1 GO:1903975                                           regulation of glial cell migration      3/84
## 1334       BP       1 GO:1903557 positive regulation of tumor necrosis factor superfamily cytokine production      5/84
## 1335       BP       1 GO:0032635                                                     interleukin-6 production      6/84
## 1336       BP       1 GO:0006909                                                                 phagocytosis      8/84
## 1337       BP       1 GO:0032609                                                  interferon-gamma production      5/84
## 1338       BP       1 GO:0042116                                                        macrophage activation      5/84
## 3318       MF       1 GO:0035014                             phosphatidylinositol 3-kinase regulator activity      3/88
## 1339       BP       1 GO:0002696                                  positive regulation of leukocyte activation      8/84
## 1340       BP       1 GO:0050702                                                 interleukin-1 beta secretion      4/84
## 1341       BP       1 GO:0060216                                                       definitive hemopoiesis      3/84
## 3319       MF       1 GO:0001614                                      purinergic nucleotide receptor activity      3/88
## 3320       MF       1 GO:0016502                                                 nucleotide receptor activity      3/88
## 1342       BP       1 GO:0071216                                         cellular response to biotic stimulus      7/84
## 3321       MF       1 GO:0004930                                          G protein-coupled receptor activity      9/88
## 1343       BP       1 GO:0050701                                                      interleukin-1 secretion      4/84
## 1344       BP       1 GO:0043032                                 positive regulation of macrophage activation      3/84
## 1345       BP       1 GO:0050901                                               leukocyte tethering or rolling      3/84
## 1346       BP       1 GO:0032729                           positive regulation of interferon-gamma production      4/84
## 1347       BP       1 GO:0006911                                                     phagocytosis, engulfment      4/84
## 1348       BP       1 GO:0070098                                         chemokine-mediated signaling pathway      4/84
## 1349       BP       1 GO:0007599                                                                   hemostasis      8/84
## 1350       BP       1 GO:0032691                         negative regulation of interleukin-1 beta production      3/84
## 3137       MF      -1 GO:0070888                                                                E-box binding      3/41
## 1351       BP       1 GO:0032692                              negative regulation of interleukin-1 production      3/84
## 1352       BP       1 GO:0001819                                   positive regulation of cytokine production      9/84
## 1353       BP       1 GO:0031349                                      positive regulation of defense response      8/84
## 1354       BP       1 GO:0050766                                          positive regulation of phagocytosis      4/84
## 3871     KEGG       1   hsa04061                Viral protein interaction with cytokine and cytokine receptor      4/46
## 1355       BP       1 GO:0043410                                          positive regulation of MAPK cascade     10/84
## 1356       BP       1 GO:0071222                                      cellular response to lipopolysaccharide      6/84
## 1357       BP       1 GO:0099024                                                 plasma membrane invagination      4/84
## 1358       BP       1 GO:1990868                                                        response to chemokine      4/84
## 1359       BP       1 GO:1990869                                               cellular response to chemokine      4/84
## 1360       BP       1 GO:0032637                                                     interleukin-8 production      4/84
## 1361       BP       1 GO:0050729                                 positive regulation of inflammatory response      5/84
## 1362       BP       1 GO:0050718                          positive regulation of interleukin-1 beta secretion      3/84
## 1363       BP       1 GO:0032675                                       regulation of interleukin-6 production      5/84
## 1364       BP       1 GO:0035590                             purinergic nucleotide receptor signaling pathway      3/84
## 1365       BP       1 GO:0050716                               positive regulation of interleukin-1 secretion      3/84
## 1366       BP       1 GO:0032103                         positive regulation of response to external stimulus      9/84
## 1367       BP       1 GO:0002831                                    regulation of response to biotic stimulus      8/84
## 1368       BP       1 GO:0002275                          myeloid cell activation involved in immune response     10/84
## 1369       BP       1 GO:0030851                                                  granulocyte differentiation      3/84
## 1370       BP       1 GO:2000403                                  positive regulation of lymphocyte migration      3/84
## 1371       BP       1 GO:0002573                                            myeloid leukocyte differentiation      6/84
## 1372       BP       1 GO:0010324                                                        membrane invagination      4/84
## 1373       BP       1 GO:0061756                              leukocyte adhesion to vascular endothelial cell      3/84
## 1374       BP       1 GO:0032649                                    regulation of interferon-gamma production      4/84
## 3872     KEGG       1   hsa04060                                       Cytokine-cytokine receptor interaction      6/46
## 3322       MF       1 GO:0005126                                                    cytokine receptor binding      6/88
## 1375       BP       1 GO:0032496                                               response to lipopolysaccharide      7/84
## 1376       BP       1 GO:1903532                                     positive regulation of secretion by cell      8/84
## 1377       BP       1 GO:0072676                                                         lymphocyte migration      4/84
## 1378       BP       1 GO:0007159                                                 leukocyte cell-cell adhesion      7/84
## 1379       BP       1 GO:0032755                              positive regulation of interleukin-6 production      4/84
## 1380       BP       1 GO:0002237                                     response to molecule of bacterial origin      7/84
## 1381       BP       1 GO:0002683                                 negative regulation of immune system process      8/84
## 1382       BP       1 GO:1905521                                           regulation of macrophage migration      3/84
## 1383       BP       1 GO:0007596                                                            blood coagulation      7/84
## 1384       BP       1 GO:0050817                                                                  coagulation      7/84
## 1385       BP       1 GO:0050706                                   regulation of interleukin-1 beta secretion      3/84
## 1386       BP       1 GO:0051047                                             positive regulation of secretion      8/84
## 1387       BP       1 GO:0042035                                  regulation of cytokine biosynthetic process      4/84
## 1388       BP       1 GO:0050764                                                   regulation of phagocytosis      4/84
## 1389       BP       1 GO:0001774                                                   microglial cell activation      3/84
## 1390       BP       1 GO:0002269                       leukocyte activation involved in inflammatory response      3/84
## 1391       BP       1 GO:0030225                                                   macrophage differentiation      3/84
## 1392       BP       1 GO:0050704                                        regulation of interleukin-1 secretion      3/84
## 3323       MF       1 GO:0000978        RNA polymerase II cis-regulatory region sequence-specific DNA binding      9/88
## 1393       BP       1 GO:0030335                                        positive regulation of cell migration      9/84
## 3324       MF       1 GO:0005525                                                                  GTP binding      8/88
## 1394       BP       1 GO:0030099                                                 myeloid cell differentiation      8/84
## 1395       BP       1 GO:0072604                                                      interleukin-6 secretion      3/84
## 3325       MF       1 GO:0032550                                                purine ribonucleoside binding      8/88
## 3326       MF       1 GO:0001883                                                    purine nucleoside binding      8/88
## 3327       MF       1 GO:0032549                                                       ribonucleoside binding      8/88
## 1396       BP       1 GO:0008217                                                 regulation of blood pressure      5/84
## 1397       BP       1 GO:0042089                                                cytokine biosynthetic process      4/84
## 1398       BP       1 GO:0042107                                                   cytokine metabolic process      4/84
## 3328       MF       1 GO:0000987                          cis-regulatory region sequence-specific DNA binding      9/88
## 1399       BP       1 GO:1902105                                      regulation of leukocyte differentiation      6/84
## 1400       BP       1 GO:0045766                                          positive regulation of angiogenesis      5/84
## 1401       BP       1 GO:2000147                                         positive regulation of cell motility      9/84
## 1402       BP       1 GO:0002363                                         alpha-beta T cell lineage commitment      2/84
## 1403       BP       1 GO:0150078                            positive regulation of neuroinflammatory response      2/84
## 1404       BP       1 GO:0033003                                           regulation of mast cell activation      3/84
## 3329       MF       1 GO:0001882                                                           nucleoside binding      8/88
## 1405       BP       1 GO:0032731                         positive regulation of interleukin-1 beta production      3/84
## 3330       MF       1 GO:0019001                                                    guanyl nucleotide binding      8/88
## 3331       MF       1 GO:0032561                                                guanyl ribonucleotide binding      8/88
## 1406       BP       1 GO:0001818                                   negative regulation of cytokine production      6/84
## 1407       BP       1 GO:0048863                                                    stem cell differentiation      6/84
## 1408       BP       1 GO:0043030                                          regulation of macrophage activation      3/84
## 1409       BP       1 GO:0002697                                        regulation of immune effector process      7/84
## 1410       BP       1 GO:0040017                                            positive regulation of locomotion      9/84
## 1411       BP       1 GO:0043369           CD4-positive or CD8-positive, alpha-beta T cell lineage commitment      2/84
## 1412       BP       1 GO:0098883                                                              synapse pruning      2/84
## 1413       BP       1 GO:0051272                           positive regulation of cellular component movement      9/84
## 3332       MF       1 GO:0070888                                                                E-box binding      3/88
## 1414       BP       1 GO:0097529                                                  myeloid leukocyte migration      5/84
## 1415       BP       1 GO:0043300                                        regulation of leukocyte degranulation      3/84
## 1416       BP       1 GO:0002244                                hematopoietic progenitor cell differentiation      5/84
## 1417       BP       1 GO:0002703                                    regulation of leukocyte mediated immunity      5/84
## 1418       BP       1 GO:0070374                                 positive regulation of ERK1 and ERK2 cascade      5/84
## 3333       MF       1 GO:0048020                                               CCR chemokine receptor binding      2/88
## 1419       BP       1 GO:1903708                                           positive regulation of hemopoiesis      5/84
## 1420       BP       1 GO:0032732                              positive regulation of interleukin-1 production      3/84
## 1421       BP       1 GO:1905517                                                         macrophage migration      3/84
## 1422       BP       1 GO:0070372                                          regulation of ERK1 and ERK2 cascade      6/84
## 1423       BP       1 GO:1903037                                   regulation of leukocyte cell-cell adhesion      6/84
## 1424       BP       1 GO:0071692                                 protein localization to extracellular region      9/84
## 1425       BP       1 GO:0045088                                         regulation of innate immune response      6/84
## 1426       BP       1 GO:0061900                                                        glial cell activation      3/84
## 1427       BP       1 GO:0010919                        regulation of inositol phosphate biosynthetic process      2/84
## 1428       BP       1 GO:0033004                                  negative regulation of mast cell activation      2/84
## 1429       BP       1 GO:0060100                              positive regulation of phagocytosis, engulfment      2/84
## 1430       BP       1 GO:1905155                                 positive regulation of membrane invagination      2/84
## 1431       BP       1 GO:0050878                                              regulation of body fluid levels      8/84
## 1432       BP       1 GO:0002761                              regulation of myeloid leukocyte differentiation      4/84
## 1433       BP       1 GO:1904018                               positive regulation of vasculature development      5/84
## 1434       BP       1 GO:0008347                                                         glial cell migration      3/84
## 1435       BP       1 GO:0043551                         regulation of phosphatidylinositol 3-kinase activity      3/84
## 1436       BP       1 GO:0072678                                                             T cell migration      3/84
## 3334       MF       1 GO:0019955                                                             cytokine binding      4/88
## 1437       BP       1 GO:0001562                                                        response to protozoan      2/84
## 1438       BP       1 GO:0042832                                                defense response to protozoan      2/84
## 1439       BP       1 GO:0060099                                       regulation of phagocytosis, engulfment      2/84
## 1440       BP       1 GO:1903978                                     regulation of microglial cell activation      2/84
## 1441       BP       1 GO:2000401                                           regulation of lymphocyte migration      3/84
## 1442       BP       1 GO:0043270                                         positive regulation of ion transport      6/84
## 1443       BP       1 GO:0042108                         positive regulation of cytokine biosynthetic process      3/84
## 1444       BP       1 GO:0070371                                                        ERK1 and ERK2 cascade      6/84
## 1445       BP       1 GO:0002790                                                            peptide secretion      9/84
## 1446       BP       1 GO:0002360                                                    T cell lineage commitment      2/84
## 1447       BP       1 GO:0042104                        positive regulation of activated T cell proliferation      2/84
## 1448       BP       1 GO:1905153                                          regulation of membrane invagination      2/84
## 1449       BP       1 GO:0002886                            regulation of myeloid leukocyte mediated immunity      3/84
## 1450       BP       1 GO:0045765                                                   regulation of angiogenesis      6/84
## 1451       BP       1 GO:0032677                                       regulation of interleukin-8 production      3/84
## 1452       BP       1 GO:0045123                                                       cellular extravasation      3/84
## 1453       BP       1 GO:0050727                                          regulation of inflammatory response      6/84
## 1454       BP       1 GO:0002577                            regulation of antigen processing and presentation      2/84
## 1455       BP       1 GO:0150146                                                    cell junction disassembly      2/84
## 1456       BP       1 GO:0006954                                                        inflammatory response      9/84
## 1457       BP       1 GO:0031663                                lipopolysaccharide-mediated signaling pathway      3/84
##            pvalue
## 1311 4.442846e-08
## 1312 5.955807e-08
## 1313 2.042035e-07
## 1314 2.762117e-07
## 1315 4.846558e-07
## 1316 5.291141e-07
## 1317 6.457953e-07
## 1318 2.297132e-06
## 1319 2.349005e-06
## 1320 3.605350e-06
## 1321 3.605350e-06
## 1322 3.637541e-06
## 1323 6.604984e-06
## 1324 9.663583e-06
## 1325 2.188166e-05
## 1326 2.555242e-05
## 3316 3.086309e-05
## 1327 3.901699e-05
## 1328 3.938836e-05
## 3317 4.223712e-05
## 1329 4.302992e-05
## 1330 4.665008e-05
## 1331 4.862740e-05
## 1332 5.573978e-05
## 1333 6.417224e-05
## 1334 6.438378e-05
## 1335 6.633435e-05
## 1336 7.418961e-05
## 1337 7.924753e-05
## 1338 8.474696e-05
## 3318 9.186960e-05
## 1339 9.207168e-05
## 1340 9.843140e-05
## 1341 1.011745e-04
## 3319 1.142974e-04
## 3320 1.142974e-04
## 1342 1.169372e-04
## 3321 1.277876e-04
## 1343 1.320556e-04
## 1344 1.498475e-04
## 1345 1.498475e-04
## 1346 1.586476e-04
## 1347 1.733017e-04
## 1348 1.733017e-04
## 1349 2.032086e-04
## 1350 2.116152e-04
## 3137 2.187705e-04
## 1351 2.478391e-04
## 1352 2.514522e-04
## 1353 2.564692e-04
## 1354 2.825185e-04
## 3871 2.918271e-04
## 1355 2.986591e-04
## 1356 3.212669e-04
## 1357 3.279249e-04
## 1358 3.279249e-04
## 1359 3.279249e-04
## 1360 3.783360e-04
## 1361 3.798024e-04
## 1362 4.321639e-04
## 1363 4.766011e-04
## 1364 4.888890e-04
## 1365 4.888890e-04
## 1366 4.971077e-04
## 1367 5.110894e-04
## 1368 5.390682e-04
## 1369 5.501749e-04
## 1370 5.501749e-04
## 1371 5.551129e-04
## 1372 5.626928e-04
## 1373 6.161675e-04
## 1374 6.754461e-04
## 3872 7.187714e-04
## 3322 7.359591e-04
## 1375 7.424577e-04
## 1376 7.846042e-04
## 1377 8.033448e-04
## 1378 8.182442e-04
## 1379 8.495298e-04
## 1380 9.655090e-04
## 1381 9.791509e-04
## 1382 1.021620e-03
## 1383 1.034691e-03
## 1384 1.082942e-03
## 1385 1.118743e-03
## 1386 1.143604e-03
## 1387 1.166745e-03
## 1388 1.166745e-03
## 1389 1.330039e-03
## 1390 1.330039e-03
## 1391 1.330039e-03
## 1392 1.330039e-03
## 3323 1.337238e-03
## 1393 1.367487e-03
## 3324 1.384455e-03
## 1394 1.431051e-03
## 1395 1.444451e-03
## 3325 1.493956e-03
## 3326 1.580520e-03
## 3327 1.580520e-03
## 1396 1.596835e-03
## 1397 1.632240e-03
## 1398 1.632240e-03
## 3328 1.632513e-03
## 1399 1.635117e-03
## 1400 1.649303e-03
## 1401 1.654766e-03
## 1402 1.677685e-03
## 1403 1.677685e-03
## 1404 1.691385e-03
## 3329 1.798052e-03
## 1405 1.824131e-03
## 3330 1.830975e-03
## 3331 1.830975e-03
## 1406 1.942791e-03
## 1407 1.942791e-03
## 1408 1.963208e-03
## 1409 2.003158e-03
## 1410 2.018912e-03
## 1411 2.042202e-03
## 1412 2.042202e-03
## 1413 2.049387e-03
## 3332 2.051128e-03
## 1414 2.052809e-03
## 1415 2.108721e-03
## 1416 2.115847e-03
## 1417 2.180288e-03
## 1418 2.180288e-03
## 3333 2.213958e-03
## 1419 2.246151e-03
## 1420 2.260772e-03
## 1421 2.260772e-03
## 1422 2.292524e-03
## 1423 2.346125e-03
## 1424 2.374893e-03
## 1425 2.400666e-03
## 1426 2.419462e-03
## 1427 2.440725e-03
## 1428 2.440725e-03
## 1429 2.440725e-03
## 1430 2.440725e-03
## 1431 2.485301e-03
## 1432 2.602808e-03
## 1433 2.748562e-03
## 1434 2.757143e-03
## 1435 2.757143e-03
## 1436 2.757143e-03
## 3334 2.788894e-03
## 1437 2.872826e-03
## 1438 2.872826e-03
## 1439 2.872826e-03
## 1440 2.872826e-03
## 1441 2.936322e-03
## 1442 3.000153e-03
## 1443 3.122514e-03
## 1444 3.132401e-03
## 1445 3.149398e-03
## 1446 3.338079e-03
## 1447 3.338079e-03
## 1448 3.338079e-03
## 1449 3.516285e-03
## 1450 3.629843e-03
## 1451 3.724031e-03
## 1452 3.724031e-03
## 1453 3.782205e-03
## 1454 3.836064e-03
## 1455 3.836064e-03
## 1456 3.902893e-03
## 1457 3.939125e-03</code></pre>
</div>
<div id="visualize-differentially-expressed-genes" class="section level2">
<h2><span class="header-section-number">1.9</span> Visualize differentially expressed genes</h2>
<p>Here we visualize DEGs with a heatmap.</p>
<pre class="r"><code>exprs_heatmap = vGene$E[rownames(sigGene),]

df &lt;- as.data.frame(colData(rse_gene)[,c(&quot;PrimaryDx&quot;)])
rownames(df) &lt;- colnames(exprs_heatmap)
colnames(df)&lt;-&quot;diagnosis&quot;

# Display heatmap
pheatmap(exprs_heatmap, cluster_rows=TRUE, show_rownames=FALSE,
         cluster_cols=TRUE, annotation_col=df)</code></pre>
<p><img src="de_analysis_speaqeasy_files/figure-html/heatmap-1.png" width="768" /></p>
<pre class="r"><code># Write heatmap to PDF
pdf(file=here(&quot;DE_analysis&quot;, &quot;pdfs&quot;, &quot;de_heatmap.pdf&quot;))
pheatmap(exprs_heatmap, cluster_rows=TRUE, show_rownames=FALSE,
         cluster_cols=TRUE, annotation_col=df)
dev.off()</code></pre>
<pre><code>## pdf 
##   3</code></pre>
</div>
</div>
<div id="reproducibility" class="section level1">
<h1><span class="header-section-number">2</span> Reproducibility</h1>
<p>This analysis report was made possible thanks to:</p>
<ul>
<li>R (<a href='#bib-R'>Kolde, 2019</a>)</li>
<li><em><a href="https://bioconductor.org/packages/3.12/BiocStyle">BiocStyle</a></em> (<a href='https://www.R-project.org/'>R Core Team, 2020</a>)</li>
<li><em><a href="https://bioconductor.org/packages/3.12/clusterProfiler">clusterProfiler</a></em> (<a href='https://github.com/rstudio/rmarkdown'>Allaire, Xie, McPherson, Luraschi, et al., 2020</a>)</li>
<li><em><a href="https://CRAN.R-project.org/package=devtools">devtools</a></em> (<a href='https://github.com/cboettig/knitcitations'>Boettiger, 2019</a>)</li>
<li><em><a href="https://bioconductor.org/packages/3.12/edgeR">edgeR</a></em></li>
<li><em><a href="https://CRAN.R-project.org/package=here">here</a></em> (<a href='http://www.nature.com/nbt/journal/v35/n4/full/nbt.3838.html'>Collado-Torres, Nellore, Kammers, Ellis, et al., 2017</a>)</li>
<li><em><a href="https://github.com/LieberInstitute/jaffelab">jaffelab</a></em> (<a href='https://github.com/LieberInstitute/jaffelab'>Collado-Torres, Jaffe, and Burke, 2019</a>)</li>
<li><em><a href="https://bioconductor.org/packages/3.12/limma">limma</a></em> (<a href='https://doi.org/10.1093/nar/gky102'>Ellis, Collado-Torres, Jaffe, and Leek, 2018</a>)</li>
<li><em><a href="https://CRAN.R-project.org/package=knitcitations">knitcitations</a></em> (<a href='http://www.bioconductor.org/packages/recount'>Collado-Torres, Nellore, Kammers, Ellis, et al., 2020</a>)</li>
<li><em><a href="https://bioconductor.org/packages/3.12/org.Hs.eg.db">org.Hs.eg.db</a></em> (<a href='https://doi.org/10.1186/1471-2105-12-449'>Frazee, Langmead, and Leek, 2011</a>)</li>
<li><em><a href="https://CRAN.R-project.org/package=pheatmap">pheatmap</a></em> (<a href='https://doi.org/10.1101/gr.254656.119'>Imada, Sanchez, Collado-Torres, Wilks, et al., 2020</a>)</li>
<li><em><a href="https://CRAN.R-project.org/package=RColorBrewer">RColorBrewer</a></em> (<a href='https://doi.org/10.1093/nar/gks042'>McCarthy, Chen, and Smyth, 2012</a>)</li>
<li><em><a href="https://bioconductor.org/packages/3.12/recount">recount</a></em></li>
<li><em><a href="https://CRAN.R-project.org/package=rmarkdown">rmarkdown</a></em></li>
<li><em><a href="https://bioconductor.org/packages/3.12/SummarizedExperiment">SummarizedExperiment</a></em> (<a href='https://bookdown.org/yihui/rmarkdown-cookbook'>Xie, Dervieux, and Riederer, 2020</a>)</li>
<li><em><a href="https://bioconductor.org/packages/3.12/voom">voom</a></em> (<a href='https://doi.org/10.1089/omi.2011.0118'>Yu, Wang, Han, and He, 2012</a>)</li>
</ul>
<p><a href="de_analysis.bib">Bibliography file</a></p>
<p>[1] J. Allaire, Y. Xie, J. McPherson, J. Luraschi, et al. <em>rmarkdown: Dynamic Documents for R</em>. R package version
2.4. 2020. &lt;URL: <a href="https://github.com/rstudio/rmarkdown" class="uri">https://github.com/rstudio/rmarkdown</a>&gt;.</p>
<p>[2] C. Boettiger. <em>knitcitations: Citations for ‘Knitr’ Markdown Files</em>. R package version 1.0.10. 2019. &lt;URL:
<a href="https://github.com/cboettig/knitcitations" class="uri">https://github.com/cboettig/knitcitations</a>&gt;.</p>
<p>[3] L. Collado-Torres, A. E. Jaffe, and E. E. Burke. <em>jaffelab: Commonly used functions by the Jaffe lab</em>. R package
version 0.99.30. 2019. &lt;URL: <a href="https://github.com/LieberInstitute/jaffelab" class="uri">https://github.com/LieberInstitute/jaffelab</a>&gt;.</p>
<p>[4] L. Collado-Torres, A. Nellore, K. Kammers, S. E. Ellis, et al. <em>Explore and download data from the recount
project</em>. <a href="https://github.com/leekgroup/recount" class="uri">https://github.com/leekgroup/recount</a> - R package version 1.15.1. 2020. DOI: 10.18129/B9.bioc.recount.
&lt;URL: <a href="http://www.bioconductor.org/packages/recount" class="uri">http://www.bioconductor.org/packages/recount</a>&gt;.</p>
<p>[5] L. Collado-Torres, A. Nellore, K. Kammers, S. E. Ellis, et al. “Reproducible RNA-seq analysis using recount2”.
In: <em>Nature Biotechnology</em> (2017). DOI: 10.1038/nbt.3838. &lt;URL:
<a href="http://www.nature.com/nbt/journal/v35/n4/full/nbt.3838.html" class="uri">http://www.nature.com/nbt/journal/v35/n4/full/nbt.3838.html</a>&gt;.</p>
<p>[6] S. E. Ellis, L. Collado-Torres, A. E. Jaffe, and J. T. Leek. “Improving the value of public RNA-seq expression
data by phenotype prediction”. In: <em>Nucl. Acids Res.</em> (2018). DOI: 10.1093/nar/gky102. &lt;URL:
<a href="https://doi.org/10.1093/nar/gky102" class="uri">https://doi.org/10.1093/nar/gky102</a>&gt;.</p>
<p>[7] A. C. Frazee, B. Langmead, and J. T. Leek. “ReCount: A multi-experiment resource of analysis-ready RNA-seq gene
count datasets”. In: <em>BMC Bioinformatics</em> (2011). DOI: 10.1186/1471-2105-12-449. &lt;URL:
<a href="https://doi.org/10.1186/1471-2105-12-449" class="uri">https://doi.org/10.1186/1471-2105-12-449</a>&gt;.</p>
<p>[8] E. Imada, D. F. Sanchez, L. Collado-Torres, C. Wilks, et al. “Recounting the FANTOM CAGE–Associated
Transcriptome”. In: <em>Genome Research</em> (2020). DOI: 10.1101/gr.254656.119. &lt;URL:
<a href="https://doi.org/10.1101/gr.254656.119" class="uri">https://doi.org/10.1101/gr.254656.119</a>&gt;.</p>
<p>[9] R. Kolde. <em>pheatmap: Pretty Heatmaps</em>. R package version 1.0.12. 2019.</p>
<p>[10] D. J. McCarthy, Y. Chen, and G. K. Smyth. “Differential expression analysis of multifactor RNA-Seq experiments
with respect to biological variation”. In: <em>Nucleic Acids Research</em> 40.10 (2012), pp. 4288-4297. DOI:
10.1093/nar/gks042.</p>
<p>[11] R Core Team. <em>R: A Language and Environment for Statistical Computing</em>. R Foundation for Statistical Computing.
Vienna, Austria, 2020. &lt;URL: <a href="https://www.R-project.org/" class="uri">https://www.R-project.org/</a>&gt;.</p>
<p>[12] Y. Xie, C. Dervieux, and E. Riederer. <em>R Markdown Cookbook</em>. ISBN 9780367563837. Boca Raton, Florida: Chapman
and Hall/CRC, 2020. &lt;URL: <a href="https://bookdown.org/yihui/rmarkdown-cookbook" class="uri">https://bookdown.org/yihui/rmarkdown-cookbook</a>&gt;.</p>
<p>[13] G. Yu, L. Wang, Y. Han, and Q. He. “clusterProfiler: an R package for comparing biological themes among gene
clusters”. In: <em>OMICS: A Journal of Integrative Biology</em> 16.5 (2012), pp. 284-287. DOI: 10.1089/omi.2011.0118.</p>
<pre class="r"><code># Time spent creating this report:
diff(c(timestart, Sys.time()))</code></pre>
<pre><code>## Time difference of 1.394572 mins</code></pre>
<pre class="r"><code># Date this report was generated
message(Sys.time())</code></pre>
<pre><code>## 2020-10-05 16:28:50</code></pre>
<pre class="r"><code># Reproducibility info
options(width = 120)
devtools::session_info()</code></pre>
<pre><code>## ─ Session info ───────────────────────────────────────────────────────────────────────────────────────────────────────
##  setting  value                       
##  version  R version 4.0.2 (2020-06-22)
##  os       Ubuntu 20.04.1 LTS          
##  system   x86_64, linux-gnu           
##  ui       X11                         
##  language (EN)                        
##  collate  en_US.UTF-8                 
##  ctype    en_US.UTF-8                 
##  tz       Etc/UTC                     
##  date     2020-10-05                  
## 
## ─ Packages ───────────────────────────────────────────────────────────────────────────────────────────────────────────
##  package              * version  date       lib source                                   
##  AnnotationDbi        * 1.51.3   2020-07-25 [1] Bioconductor                             
##  askpass                1.1      2019-01-13 [2] RSPM (R 4.0.0)                           
##  assertthat             0.2.1    2019-03-21 [2] RSPM (R 4.0.0)                           
##  backports              1.1.10   2020-09-15 [2] RSPM (R 4.0.2)                           
##  base64enc              0.1-3    2015-07-28 [2] RSPM (R 4.0.0)                           
##  bibtex                 0.4.2.3  2020-09-19 [1] RSPM (R 4.0.2)                           
##  Biobase              * 2.49.1   2020-09-03 [1] Bioconductor                             
##  BiocFileCache          1.13.1   2020-08-04 [1] Bioconductor                             
##  BiocGenerics         * 0.35.4   2020-06-04 [1] Bioconductor                             
##  BiocManager          * 1.30.10  2019-11-16 [2] CRAN (R 4.0.2)                           
##  BiocParallel           1.23.2   2020-07-06 [1] Bioconductor                             
##  BiocStyle            * 2.17.1   2020-09-24 [1] Bioconductor                             
##  biomaRt                2.45.3   2020-09-21 [1] Bioconductor                             
##  Biostrings             2.57.2   2020-06-09 [1] Bioconductor                             
##  bit                    4.0.4    2020-08-04 [1] RSPM (R 4.0.2)                           
##  bit64                  4.0.5    2020-08-30 [1] RSPM (R 4.0.2)                           
##  bitops                 1.0-6    2013-08-17 [1] RSPM (R 4.0.0)                           
##  blob                   1.2.1    2020-01-20 [1] RSPM (R 4.0.0)                           
##  bookdown               0.20     2020-06-23 [1] RSPM (R 4.0.2)                           
##  BSgenome               1.57.6   2020-09-02 [1] Bioconductor                             
##  bumphunter             1.31.0   2020-04-28 [1] Bioconductor                             
##  callr                  3.4.4    2020-09-07 [2] RSPM (R 4.0.2)                           
##  checkmate              2.0.0    2020-02-06 [1] RSPM (R 4.0.0)                           
##  cli                    2.0.2    2020-02-28 [2] RSPM (R 4.0.0)                           
##  cluster                2.1.0    2019-06-19 [3] CRAN (R 4.0.2)                           
##  clusterProfiler      * 3.17.3   2020-09-10 [1] Bioconductor                             
##  codetools              0.2-16   2018-12-24 [3] CRAN (R 4.0.2)                           
##  colorspace             1.4-1    2019-03-18 [1] RSPM (R 4.0.0)                           
##  cowplot                1.1.0    2020-09-08 [1] RSPM (R 4.0.2)                           
##  crayon                 1.3.4    2017-09-16 [2] RSPM (R 4.0.0)                           
##  curl                   4.3      2019-12-02 [2] RSPM (R 4.0.0)                           
##  data.table             1.13.0   2020-07-24 [1] RSPM (R 4.0.2)                           
##  DBI                    1.1.0    2019-12-15 [1] RSPM (R 4.0.0)                           
##  dbplyr                 1.4.4    2020-05-27 [1] RSPM (R 4.0.0)                           
##  DelayedArray           0.15.15  2020-10-03 [1] Bioconductor                             
##  derfinder              1.23.1   2020-05-10 [1] Bioconductor                             
##  derfinderHelper        1.23.1   2020-05-10 [1] Bioconductor                             
##  desc                   1.2.0    2018-05-01 [2] RSPM (R 4.0.0)                           
##  devtools             * 2.3.2    2020-09-18 [2] RSPM (R 4.0.2)                           
##  digest                 0.6.25   2020-02-23 [2] RSPM (R 4.0.0)                           
##  DO.db                  2.9      2020-10-05 [1] Bioconductor                             
##  doRNG                  1.8.2    2020-01-27 [1] RSPM (R 4.0.0)                           
##  DOSE                   3.15.1   2020-09-10 [1] Bioconductor                             
##  downloader             0.4      2015-07-09 [1] RSPM (R 4.0.0)                           
##  dplyr                  1.0.2    2020-08-18 [1] RSPM (R 4.0.2)                           
##  edgeR                * 3.31.4   2020-06-10 [1] Bioconductor                             
##  ellipsis               0.3.1    2020-05-15 [2] RSPM (R 4.0.0)                           
##  enrichplot             1.9.3    2020-09-11 [1] Bioconductor                             
##  evaluate               0.14     2019-05-28 [2] RSPM (R 4.0.0)                           
##  fansi                  0.4.1    2020-01-08 [2] RSPM (R 4.0.0)                           
##  farver                 2.0.3    2020-01-16 [1] RSPM (R 4.0.0)                           
##  fastmatch              1.1-0    2017-01-28 [1] RSPM (R 4.0.0)                           
##  fgsea                  1.15.2   2020-09-15 [1] Bioconductor                             
##  foreach                1.5.0    2020-03-30 [1] RSPM (R 4.0.0)                           
##  foreign                0.8-80   2020-05-24 [3] CRAN (R 4.0.2)                           
##  Formula                1.2-3    2018-05-03 [1] RSPM (R 4.0.0)                           
##  fs                     1.5.0    2020-07-31 [2] RSPM (R 4.0.2)                           
##  generics               0.0.2    2018-11-29 [1] RSPM (R 4.0.0)                           
##  GenomeInfoDb         * 1.25.11  2020-09-03 [1] Bioconductor                             
##  GenomeInfoDbData       1.2.3    2020-10-05 [1] Bioconductor                             
##  GenomicAlignments      1.25.3   2020-06-09 [1] Bioconductor                             
##  GenomicFeatures        1.41.3   2020-09-03 [1] Bioconductor                             
##  GenomicFiles           1.25.0   2020-04-27 [1] Bioconductor                             
##  GenomicRanges        * 1.41.6   2020-08-12 [1] Bioconductor                             
##  GEOquery               2.57.0   2020-04-27 [1] Bioconductor                             
##  ggforce                0.3.2    2020-06-23 [1] RSPM (R 4.0.2)                           
##  ggnewscale             0.4.3    2020-08-27 [1] RSPM (R 4.0.2)                           
##  ggplot2                3.3.2    2020-06-19 [1] RSPM (R 4.0.1)                           
##  ggplotify              0.0.5    2020-03-12 [1] RSPM (R 4.0.0)                           
##  ggraph                 2.0.3    2020-05-20 [1] RSPM (R 4.0.2)                           
##  ggrepel                0.8.2    2020-03-08 [1] RSPM (R 4.0.2)                           
##  ggridges               0.5.2    2020-01-12 [1] RSPM (R 4.0.0)                           
##  glue                   1.4.2    2020-08-27 [2] RSPM (R 4.0.2)                           
##  GO.db                  3.11.4   2020-10-05 [1] Bioconductor                             
##  googledrive            1.0.1    2020-05-05 [1] RSPM (R 4.0.0)                           
##  GOSemSim               2.15.2   2020-09-04 [1] Bioconductor                             
##  graphlayouts           0.7.0    2020-04-25 [1] RSPM (R 4.0.2)                           
##  gridExtra              2.3      2017-09-09 [1] RSPM (R 4.0.0)                           
##  gridGraphics           0.5-0    2020-02-25 [1] RSPM (R 4.0.0)                           
##  gtable                 0.3.0    2019-03-25 [1] RSPM (R 4.0.0)                           
##  here                 * 0.1      2017-05-28 [1] RSPM (R 4.0.0)                           
##  Hmisc                  4.4-1    2020-08-10 [1] RSPM (R 4.0.2)                           
##  hms                    0.5.3    2020-01-08 [1] RSPM (R 4.0.0)                           
##  htmlTable              2.1.0    2020-09-16 [1] RSPM (R 4.0.2)                           
##  htmltools              0.5.0    2020-06-16 [2] RSPM (R 4.0.1)                           
##  htmlwidgets            1.5.1    2019-10-08 [2] RSPM (R 4.0.0)                           
##  httr                   1.4.2    2020-07-20 [2] RSPM (R 4.0.2)                           
##  igraph                 1.2.5    2020-03-19 [1] RSPM (R 4.0.0)                           
##  IRanges              * 2.23.10  2020-06-13 [1] Bioconductor                             
##  iterators              1.0.12   2019-07-26 [1] RSPM (R 4.0.0)                           
##  jaffelab             * 0.99.30  2020-10-05 [1] Github (LieberInstitute/jaffelab@42637ff)
##  jpeg                   0.1-8.1  2019-10-24 [1] RSPM (R 4.0.0)                           
##  jsonlite               1.7.1    2020-09-07 [2] RSPM (R 4.0.2)                           
##  knitcitations        * 1.0.10   2019-09-15 [1] RSPM (R 4.0.0)                           
##  knitr                  1.30     2020-09-22 [2] RSPM (R 4.0.2)                           
##  lattice                0.20-41  2020-04-02 [3] CRAN (R 4.0.2)                           
##  latticeExtra           0.6-29   2019-12-19 [1] RSPM (R 4.0.0)                           
##  lifecycle              0.2.0    2020-03-06 [2] RSPM (R 4.0.0)                           
##  limma                * 3.45.14  2020-09-12 [1] Bioconductor                             
##  locfit                 1.5-9.4  2020-03-25 [1] RSPM (R 4.0.0)                           
##  lubridate              1.7.9    2020-06-08 [1] RSPM (R 4.0.2)                           
##  magrittr               1.5      2014-11-22 [2] RSPM (R 4.0.0)                           
##  MASS                   7.3-51.6 2020-04-26 [3] CRAN (R 4.0.2)                           
##  Matrix                 1.2-18   2019-11-27 [3] CRAN (R 4.0.2)                           
##  MatrixGenerics       * 1.1.3    2020-09-30 [1] Bioconductor                             
##  matrixStats          * 0.57.0   2020-09-25 [1] RSPM (R 4.0.2)                           
##  memoise                1.1.0    2017-04-21 [2] RSPM (R 4.0.0)                           
##  munsell                0.5.0    2018-06-12 [1] RSPM (R 4.0.0)                           
##  nnet                   7.3-14   2020-04-26 [3] CRAN (R 4.0.2)                           
##  openssl                1.4.3    2020-09-18 [2] RSPM (R 4.0.2)                           
##  org.Hs.eg.db         * 3.11.4   2020-10-05 [1] Bioconductor                             
##  pheatmap             * 1.0.12   2019-01-04 [1] RSPM (R 4.0.0)                           
##  pillar                 1.4.6    2020-07-10 [2] RSPM (R 4.0.2)                           
##  pkgbuild               1.1.0    2020-07-13 [2] RSPM (R 4.0.2)                           
##  pkgconfig              2.0.3    2019-09-22 [2] RSPM (R 4.0.0)                           
##  pkgload                1.1.0    2020-05-29 [2] RSPM (R 4.0.0)                           
##  plyr                   1.8.6    2020-03-03 [1] RSPM (R 4.0.2)                           
##  png                    0.1-7    2013-12-03 [1] RSPM (R 4.0.0)                           
##  polyclip               1.10-0   2019-03-14 [1] RSPM (R 4.0.0)                           
##  prettyunits            1.1.1    2020-01-24 [2] RSPM (R 4.0.0)                           
##  processx               3.4.4    2020-09-03 [2] RSPM (R 4.0.2)                           
##  progress               1.2.2    2019-05-16 [1] RSPM (R 4.0.0)                           
##  ps                     1.3.4    2020-08-11 [2] RSPM (R 4.0.2)                           
##  purrr                  0.3.4    2020-04-17 [2] RSPM (R 4.0.0)                           
##  qvalue                 2.21.0   2020-04-27 [1] Bioconductor                             
##  R6                     2.4.1    2019-11-12 [2] RSPM (R 4.0.0)                           
##  rafalib              * 1.0.0    2015-08-09 [1] RSPM (R 4.0.0)                           
##  rappdirs               0.3.1    2016-03-28 [1] RSPM (R 4.0.0)                           
##  RColorBrewer         * 1.1-2    2014-12-07 [1] RSPM (R 4.0.0)                           
##  Rcpp                   1.0.5    2020-07-06 [2] RSPM (R 4.0.2)                           
##  RCurl                  1.98-1.2 2020-04-18 [1] RSPM (R 4.0.0)                           
##  readr                  1.3.1    2018-12-21 [1] RSPM (R 4.0.2)                           
##  recount              * 1.15.1   2020-05-12 [1] Bioconductor                             
##  RefManageR             1.2.12   2019-04-03 [1] RSPM (R 4.0.0)                           
##  remotes                2.2.0    2020-07-21 [1] RSPM (R 4.0.2)                           
##  rentrez                1.2.2    2019-05-02 [1] RSPM (R 4.0.0)                           
##  reshape2               1.4.4    2020-04-09 [1] RSPM (R 4.0.2)                           
##  rlang                  0.4.7    2020-07-09 [2] RSPM (R 4.0.2)                           
##  rmarkdown              2.4      2020-09-30 [1] RSPM (R 4.0.2)                           
##  rngtools               1.5      2020-01-23 [1] RSPM (R 4.0.0)                           
##  rpart                  4.1-15   2019-04-12 [3] CRAN (R 4.0.2)                           
##  rprojroot              1.3-2    2018-01-03 [2] RSPM (R 4.0.0)                           
##  Rsamtools              2.5.3    2020-06-26 [1] Bioconductor                             
##  RSQLite                2.2.1    2020-09-30 [1] RSPM (R 4.0.2)                           
##  rstudioapi             0.11     2020-02-07 [2] RSPM (R 4.0.0)                           
##  rtracklayer            1.49.5   2020-08-07 [1] Bioconductor                             
##  rvcheck                0.1.8    2020-03-01 [1] RSPM (R 4.0.0)                           
##  S4Vectors            * 0.27.13  2020-09-18 [1] Bioconductor                             
##  scales                 1.1.1    2020-05-11 [1] RSPM (R 4.0.0)                           
##  scatterpie             0.1.5    2020-09-09 [1] RSPM (R 4.0.2)                           
##  segmented              1.2-0    2020-06-23 [1] RSPM (R 4.0.2)                           
##  sessioninfo            1.1.1    2018-11-05 [1] RSPM (R 4.0.0)                           
##  shadowtext             0.0.7    2019-11-06 [1] RSPM (R 4.0.0)                           
##  statmod                1.4.34   2020-02-17 [1] RSPM (R 4.0.0)                           
##  stringi                1.5.3    2020-09-09 [2] RSPM (R 4.0.2)                           
##  stringr                1.4.0    2019-02-10 [2] RSPM (R 4.0.0)                           
##  SummarizedExperiment * 1.19.9   2020-10-01 [1] Bioconductor                             
##  survival               3.1-12   2020-04-10 [3] CRAN (R 4.0.2)                           
##  testthat               2.3.2    2020-03-02 [2] RSPM (R 4.0.0)                           
##  tibble                 3.0.3    2020-07-10 [2] RSPM (R 4.0.2)                           
##  tidygraph              1.2.0    2020-05-12 [1] RSPM (R 4.0.2)                           
##  tidyr                  1.1.2    2020-08-27 [1] RSPM (R 4.0.2)                           
##  tidyselect             1.1.0    2020-05-11 [1] RSPM (R 4.0.0)                           
##  tweenr                 1.0.1    2018-12-14 [1] RSPM (R 4.0.2)                           
##  usethis              * 1.6.3    2020-09-17 [1] RSPM (R 4.0.2)                           
##  VariantAnnotation      1.35.3   2020-06-26 [1] Bioconductor                             
##  vctrs                  0.3.4    2020-08-29 [2] RSPM (R 4.0.2)                           
##  viridis                0.5.1    2018-03-29 [1] RSPM (R 4.0.0)                           
##  viridisLite            0.3.0    2018-02-01 [1] RSPM (R 4.0.0)                           
##  withr                  2.3.0    2020-09-22 [2] RSPM (R 4.0.2)                           
##  xfun                   0.18     2020-09-29 [2] RSPM (R 4.0.2)                           
##  XML                    3.99-0.5 2020-07-23 [1] RSPM (R 4.0.2)                           
##  xml2                   1.3.2    2020-04-23 [2] RSPM (R 4.0.0)                           
##  XVector                0.29.3   2020-06-25 [1] Bioconductor                             
##  yaml                   2.2.1    2020-02-01 [2] RSPM (R 4.0.0)                           
##  zlibbioc               1.35.0   2020-04-27 [1] Bioconductor                             
## 
## [1] /usr/local/lib/R/host-site-library
## [2] /usr/local/lib/R/site-library
## [3] /usr/local/lib/R/library</code></pre>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("show" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": {
      styles: {
        ".MathJax_Display": {
           "text-align": "center",
           padding: "0px 150px 0px 65px",
           margin: "0px 0px 0.5em"
        },
        "@media screen and (max-width: 991px)": {
            ".MathJax_Display": {
               "text-align": "center",
               padding: "0 0 0 0"
            }
         }
      }
    }
  });
</script>
<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
