---
title: SPEAQeasy Example Analysis
author: 
  - name: Nicholas J. Eagles
    affiliation:
    - &libd Lieber Institute for Brain Development, Johns Hopkins Medical Campus
  - name: Leonardo Collado-Torres
    affiliation:
    - *libd
    - &ccb Center for Computational Biology, Johns Hopkins University
    email: lcolladotor@gmail.com
output: 
  BiocStyle::html_document:
    toc: true
    toc_float: true
    toc_depth: 2
    code_folding: show
---

# Analysis

A guided example showing how processed results from the RNAseq pipeline [SPEAQeasy](https://github.com/LieberInstitute/SPEAQeasy) can be used in differential expression analyses and visualization. We also show how existing genotype information for a set of samples can be combined with SPEAQeasy results to resolve any identity issues, as can emerge during sequencing.

* [Downloading the example data](prepare_data.html)
* [Run SPEAQeasy on your setup](run_speaqeasy.html)
* [Resolve identity issues and attach phenotype data](swap_speaqeasy.html)
* [Perform differential expression and visualize results](de_analysis_speaqeasy.html)

# Cite `SPEAQeasy`

We hope that [`SPEAQeasy`](http://research.libd.org/SPEAQeasy/) will be useful for your research. Please use the following information to cite the package and the overall approach. Thank you!

```
@article {Eagles2021,
	author = {Eagles, Nicholas J. and Burke, Emily E. and Leonard, Jabob and Barry, Brianna K. and Stolz, Joshua M. and Huuki, Louise and Phan, BaDoi N. and Larrios Serrato, Violeta and Guti{\'e}rrez-Mill{\'a}n, Everardo and Aguilar-Ordo{\~n}ez, Israel and Jaffe, Andrew E. and Collado-Torres, Leonardo},
	title = {SPEAQeasy: a scalable pipeline for expression analysis and quantification for R/bioconductor-powered RNA-seq analyses},
	year = {2021},
	doi = {10.1186/s12859-021-04142-3},
	publisher = {Cold Spring Harbor Laboratory},
	URL = {https://doi.org/10.1186/s12859-021-04142-3},
	journal = {BMC Bioinformatics}
}
```


This is a project by the [R/Bioconductor-powered Team Data Science](https://lcolladotor.github.io/bioc_team_ds) at the [Lieber Institute for Brain Development](https://www.libd.org/).

|     |     |
|-----|-----|
| <a href="https://lcolladotor.github.io/bioc_team_ds"><img src="http://lcolladotor.github.io/img/Leo_transparente.png" width="250px"></a> | <a href="https://www.libd.org/"><img src="http://lcolladotor.github.io/img/LIBD_logo.jpg" width="250px"></a> |


# References

```{r 'setup', echo = FALSE, warning = FALSE, message = FALSE}
## Bib setup
timestart = Sys.time()
library("knitcitations")
library("BiocStyle")
library("devtools")

## Load knitcitations with a clean bibliography
cleanbib()
cite_options(
    hyperlink = "to.doc",
    citation_format = "text",
    style = "html"
)

## Write bibliography information
bibs <- c(
    R = citation(),
    biocStyle = citation("BiocStyle"),
    devtools = citation("devtools"),
    clusterProfiler = citation("clusterProfiler"),
    edgeR = citation("edgeR")[1],
    getopt = citation("getopt"),
    here = citation("here"),
    jaffelab = citation("jaffelab"),
    knitcitations = citation("knitcitations"),
    limma = citation("limma"),
    org.Hs.eg.db = citation("org.Hs.eg.db"),
    pheatmap = citation("pheatmap"),
    RColorBrewer = citation("RColorBrewer"),
    recount = citation("recount")[1],
    rmarkdown = citation("rmarkdown")[1],
    SummarizedExperiment = citation("SummarizedExperiment"),
    tidyr = citation("tidyr"),
    VariantAnnotation = citation("VariantAnnotation"),
    voom = RefManageR::BibEntry(
        "article",
        key = "voom",
        author = "CW Law and Y Chen and W Shi and GK Smyth",
        year = "2014",
        title = "Voom: precision weights unlock linear model analysis tools for RNA-seq read counts",
        journal = "Genome Biology",
        volume = "15",
        pages = "R29"
    )
)

write.bibtex(bibs, file = "index.bib")
bib <- read.bibtex("index.bib")

## Assign short names
names(bib) <- names(bibs)
```

The analysis was possible thanks to R and many of its packages:

* R `r citep(bib[['R']])`
* `r Biocpkg('BiocStyle')` `r citep(bib[['biocStyle']])`
* `r Biocpkg('clusterProfiler')` `r citep(bib[['clusterProfiler']])`
* `r CRANpkg('devtools')` `r citep(bib[['devtools']])`
* `r Biocpkg('edgeR')` `r citep(bib[['edgeR']])`
* `r CRANpkg('getopt')` `r citep(bib[['getopt']])`
* `r CRANpkg('here')` `r citep(bib[['here']])`
* `r Biocpkg('limma')` `r citep(bib[['limma']])`
* `r CRANpkg('jaffelab')` `r citep(bib[['jaffelab']])`
* `r CRANpkg('knitcitations')` `r citep(bib[['knitcitations']])`
* `r Biocpkg('org.Hs.eg.db')` `r citep(bib[['org.Hs.eg.db']])`
* `r CRANpkg('pheatmap')` `r citep(bib[['pheatmap']])`
* `r CRANpkg('RColorBrewer')` `r citep(bib[['RColorBrewer']])`
* `r Biocpkg('recount')` `r citep(bib[['recount']])`
* `r CRANpkg('rmarkdown')` `r citep(bib[['rmarkdown']])`
* `r Biocpkg('SummarizedExperiment')` `r citep(bib[['SummarizedExperiment']])`
* `r CRANpkg('tidyr')` `r citep(bib[['tidyr']])`
* `r Biocpkg('VariantAnnotation')` `r citep(bib[['VariantAnnotation']])`
* `r Biocpkg('voom')` `r citep(bib[['voom']])`

[Bibliography file](index.bib)

```{r bibliography, results='asis', echo=FALSE, warning = FALSE}
# Print bibliography
bibliography()
```

```{r reproducibility}
# Time spent creating this report:
diff(c(timestart, Sys.time()))

# Date this report was generated
message(Sys.time())

# Reproducibility info
options(width = 120)
sessioninfo::session_info()
```
